---
title: "mobster"
author: "Giulio Caravagna"
date: "`r format(Sys.time(), '%d %B, %Y')`"
institute: "Institute for Cancer Research"
email: "giulio.caravagna@icr.ac.uk"
opengraph:
  twitter:
    card: summary
    creator: "@gcaravagna"
output: rmarkdown::github_document
vignette: >
  %\VignetteIndexEntry{mobster}
  %\VignetteEngine{knitr::rmarkdown}
  %\VignetteEncoding{UTF-8}
---

```{r, include = FALSE}
knitr::opts_chunk$set(
  collapse = TRUE,
  comment = "#>"
)
```

<center>
<a href="https://caravagn.github.io/mobster"><img src="https://caravagn.github.io/mobster/reference/figures/logo.png" width=77px height=91px></img></a>
</center>
<br>

`mobster` implements a Dirichlet finite mixture model to detect ongoing positive subclonal selection from cancer genome sequencing data. The algorithm works best with high-resolution whole-genome sequencing data  (e.g., WGS `>100x`). The models performs a deconvolution of the site/ allele frequency spectrum of mutation data (the _signal_), and looks for models with `k+1` mixture components to fit the data (`k` subclones).

```{r, include = FALSE, warning=FALSE, message=FALSE, echo=F}
require(dplyr)
require(tidyr)
require(ggplot2)
require(RColorBrewer)

# Marc's simulation
load('simulationdata.Rdata')

# Fit at generation 17
inp = df %>% filter(d == 17, VAF > 0.05)
x = mobster::mobster_fit(inp, 
                         K = 2, 
                         tail = T, 
                         samples = 1, 
                         parallel = F, 
                         seed = 123, description = "MOBSTER fit")

library(ggmuller)

mydf = df %>% as_tibble() %>%
  rename(Generation = d, Identity = cloneid) %>%
  mutate(Identity = ifelse(Identity == 1, 'Ancestral', 'Subclone')) %>%
  group_by(Generation, Identity) %>%
  summarise(Population = n()) 

Muller_df <- ggmuller::get_Muller_df(data.frame(Parent = 'Ancestral', Identity = 'Subclone'), mydf)
```
```{r, fig.width=8, fig.height=3, echo=FALSE, warning=FALSE,message=FALSE}
cowplot::plot_grid(
  ggmuller::Muller_pop_plot(Muller_df) + 
    mobster:::my_ggplot_theme() +
    guides(fill = guide_legend('Clone')) +
    labs(title = 'Population size (simulation)', subtitle = "Muller plot (ggmuller)") +
    scale_fill_manual(values = alpha(brewer.pal(2, "Set1"), alpha = .7)) +
    geom_vline(xintercept = 17, linetype = 'dashed', size = .3),
  mobster::plot.dbpmm(x$best) + labs(caption = NULL)
)
```

The plot shows the fit (right) of a simulated subclonal expansion (left, Muller plot with [ggmuller](https://cran.r-project.org/web/packages/ggmuller/vignettes/ggmuller.html)); `C2`, the subclone at `~30%` allelic frequency is outgrowing an ancestral clonal population `C1`, at `~50%` allelic frequency (heterozygous mutations). Their dynamics are consistent with what we [expect from the interplay of positive selection between clones and neutral evolution within each clone](https://caravagn.github.io/mobster/articles/Example_tumour_simulation.html).

Inspired from both _mathematical modelling of evolutionary processes_ and _Machine Learning_, the  signal is modeled as mixture density with two types of distributions:

* `k` Betas to capture the peaks of alleles raising up in frequency in different clones (subclones enjoying positive selection, and the clonal cluster);
* `1` Pareto Type-I power law to model within-clone neutral dynamics, which is the distribution predicted by theoretical Population Genetics.

`mobster` fits can be computed via _moment-matching_ (default)  or _maximum-likelihood_, the former being much faster Model selection for the number of components can be done with multiple likelihood-based scores such as the BIC, and its entropy-based extensions ICL and reICL, a new variation to ICL with reduced-entropy.

S3 objects are defined to perform easy visualization of the data and aid comparison of different fits; _parametric_ and _non-parametric bootstrap_ routines are also available to assess the confidence of each parameter (bootstrap quantiles) and the model (overall model frequency).

This is a _model-based_ approach to analyse cancer data, meaning that a power law tail is used to integrate evolutionary dynamics in this traditional clustering problem. Results from `mobster` deconvolution can be used to reconstruct the clonal architecture of a tumour (_subclonal deconvolution_) and identify patterns of functional heterogeneity (subclones under positive selection). 

A number of [vignettes](https://caravagn.github.io/mobster/articles/) are available to help you using `mobster`; for a set of real case studies check out the Supplementary Data repository hosted at the [Sottoriva Lab Github page](https://github.com/sottorivalab)


