<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>1. Introduction • mobster</title>
<!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png">
<link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png">
<link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png">
<link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png">
<link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png">
<link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png">
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="1. Introduction">
<meta property="og:description" content="mobster">
<meta property="og:image" content="caravagn.github.io/mobster/logo.png">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">mobster</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.1.1</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../index.html">
    <span class="fas fa fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../articles/mobster.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/a1_introduction.html">1. Introduction</a>
    </li>
    <li>
      <a href="../articles/a2_plotting.html">2. Plotting fits</a>
    </li>
    <li>
      <a href="../articles/a3_bootstrap.html">3. Confidence estimation by the bootstrap</a>
    </li>
    <li>
      <a href="../articles/a4_popgen.html">4. Population Genetics statistics</a>
    </li>
    <li>
      <a href="../articles/a5_dnds.html">5. dN/dS statistics</a>
    </li>
    <li>
      <a href="../articles/a6_CloneTrees.html">6. Clone trees from clusters</a>
    </li>
    <li>
      <a href="../articles/a7_Example_tumour_simulation.html">7. Visualizing subclonal expansions</a>
    </li>
  </ul>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/caravagn/mobster/">
    <span class="fab fa fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>1. Introduction</h1>
                        <h4 class="author">Giulio Caravagna</h4>
            
            <h4 class="date">13 September, 2020</h4>
      
      <small class="dont-index">Source: <a href="https://github.com/caravagn/mobstervignettes/a1_introduction.Rmd"><code>vignettes/a1_introduction.Rmd</code></a></small>
      <div class="hidden name"><code>a1_introduction.Rmd</code></div>

    </div>

    
    
<div class="sourceCode" id="cb1"><pre class="downlit">
<span class="fu"><a href="https://rdrr.io/r/base/library.html">library</a></span>(<span class="kw"><a href="https://github.com/caravagn/mobster">mobster</a></span>)
<span class="fu"><a href="https://rdrr.io/r/base/library.html">library</a></span>(<span class="kw"><a href="https://tidyr.tidyverse.org">tidyr</a></span>)
<span class="fu"><a href="https://rdrr.io/r/base/library.html">library</a></span>(<span class="kw"><a href="https://dplyr.tidyverse.org">dplyr</a></span>)
</pre></div>
<div id="input-data-for-mobster" class="section level1">
<h1 class="hasAnchor">
<a href="#input-data-for-mobster" class="anchor"></a>Input data for <code>mobster</code>
</h1>
<p>You can run a MOBSTER analysis if, for a set of input mutations (SNVs, indels etc.), you have available VAF or CCF data. The input data can be loaded using different input formats.</p>
<p>For VAF values you can use:</p>
<ul>
<li>a <code>data.frame</code> (or, <code>tibble</code>) with a column named <code>VAF</code> whose values are numerical <span class="math inline">\(0&lt;x&lt;1\)</span>, without <code>NA</code> entries and columns named any of those: <code>cluster</code>, <code>Tail</code>, <code>C1</code>, <code>C2</code>, <code>C3</code>,…;</li>
<li>a VCF file that must contain at least a column with the total depth of sequencing and the number of reads with the variants. In this case you need first to use function <code>load_VCF</code> and load the VCF content, and then proceed using your data as a <code>data.frame</code>.</li>
</ul>
<p>For CCF values you can only use the a <code>data.frame</code> format. Importantly, you have to store the CCF values in a column again named <code>VAF</code>, which must follow the same convention of a VAF column (i.e., range of values). Since CCF values usually peak at around <code>1.0</code> for clonal mutations (i.e., present in 100% of the input cells), we suggest to adjust standard CCF estimates dividing them by <code>0.5</code> in order to reflect the peak of an <em>heterozygous clonal mutation</em> at 50% VAF for a 100% pure bulk sample.</p>
<p><strong>Example dataset</strong>. Diploid mutations from sample <code>LU4</code> of the <a href="http://genome.kaist.ac.kr/">Comprehensive Omics Archive of Lung Adenocarcinoma</a> are available in the package under the name <code>LU4_lung_sample</code>. The available object is the results of an analysis with <code>mobster</code>, and the input mutation data is stored inside the object.</p>
<div class="sourceCode" id="cb2"><pre class="downlit">
<span class="co"># Example dataset LU4_lung_sample, downloaded from http://genome.kaist.ac.kr/</span>
<span class="fu"><a href="https://rdrr.io/r/base/print.html">print</a></span>(<span class="kw">mobster</span>::<span class="kw"><a href="../reference/LU4_lung_sample.html">LU4_lung_sample</a></span><span class="op">$</span><span class="kw">best</span><span class="op">$</span><span class="kw">data</span>)
<span class="co">#&gt; # A tibble: 1,282 x 12</span>
<span class="co">#&gt;      Key Callers t_alt_count t_ref_count Variant_Classif…    DP    VAF chr  </span>
<span class="co">#&gt;    &lt;int&gt; &lt;chr&gt;         &lt;int&gt;       &lt;int&gt; &lt;chr&gt;            &lt;int&gt;  &lt;dbl&gt; &lt;chr&gt;</span>
<span class="co">#&gt;  1   370 mutect…          30          95 intergenic         125 0.24   chr2 </span>
<span class="co">#&gt;  2   371 mutect…          10         105 intergenic         115 0.0870 chr2 </span>
<span class="co">#&gt;  3   372 mutect…          40         108 intronic           148 0.270  chr2 </span>
<span class="co">#&gt;  4   373 mutect…          20         101 intronic           121 0.165  chr2 </span>
<span class="co">#&gt;  5   374 mutect…          39          89 intergenic         128 0.305  chr2 </span>
<span class="co">#&gt;  6   375 mutect…          41         120 intronic           161 0.255  chr2 </span>
<span class="co">#&gt;  7   376 mutect…          43          93 intronic           136 0.316  chr2 </span>
<span class="co">#&gt;  8   377 mutect…          22          95 intergenic         117 0.188  chr2 </span>
<span class="co">#&gt;  9   378 mutect            7         104 intergenic         111 0.0631 chr2 </span>
<span class="co">#&gt; 10   379 mutect…          35          93 intergenic         128 0.273  chr2 </span>
<span class="co">#&gt; # … with 1,272 more rows, and 4 more variables: from &lt;chr&gt;, ref &lt;chr&gt;,</span>
<span class="co">#&gt; #   alt &lt;chr&gt;, cluster &lt;chr&gt;</span>
</pre></div>
<p>Other datasets are available through the <code>data</code> command.</p>
<div id="driver-annotations" class="section level2">
<h2 class="hasAnchor">
<a href="#driver-annotations" class="anchor"></a>Driver annotations</h2>
<p>In the context of subclonal deconvolution we are often interested in linking “driver” events to clonal expansions. Since <code>mobster</code> works with somatic mutations data, it is possible to annotate the status of “driver mutation” in the input data; doing so, the drivers will be reported in some visualisations of the tool, but will not influence any of the computation carried out in <code>mobster</code>.</p>
<p>The annotate one or more <em>driver mutations</em> you need to include in your column 2 extra columns:</p>
<ul>
<li>
<code>is_driver</code>, a boolean <code>TRUE/FALSE</code> flag;</li>
<li>
<code>driver_label</code>, a character string that will be used as label in any visualisation that uses drivers.</li>
</ul>
</div>
</div>
<div id="generating-random-models-and-data" class="section level1">
<h1 class="hasAnchor">
<a href="#generating-random-models-and-data" class="anchor"></a>Generating random models and data</h1>
<p>You can sample a random dataset with the <code>random_dataset</code> function, setting:</p>
<ul>
<li>the number of mutations (<code>n</code>) and Beta components (<code>k</code>, subclones) to generate;</li>
<li>ranges and constraints on the size of the components;</li>
<li>ranges for the mean and variance of the Beta components.</li>
</ul>
<p>The variance of the Betas is defined as <span class="math inline">\(u/B\)</span> where <span class="math inline">\(u \sim U[0,1]\)</span>, and <span class="math inline">\(B\)</span> is the input parameter <code>Beta_variance_scaling</code>. Roughly, values of <code>Beta_variance_scaling</code> on the order of <code>1000</code> give low variance and sharp peaked data distributions. Values on the order of <code>100</code> give much wider distributions.</p>
<div class="sourceCode" id="cb3"><pre class="downlit">
<span class="kw">dataset</span> <span class="op">=</span> <span class="fu"><a href="../reference/random_dataset.html">random_dataset</a></span>(
  seed = <span class="fl">123456789</span>, 
  Beta_variance_scaling = <span class="fl">100</span>    <span class="co"># variance ~ U[0, 1]/Beta_variance_scaling</span>
  )
</pre></div>
<p>A list with 3 components is returned, which contains the actual data, sampled parameters of the generative model, and a plot of the data.</p>
<p>In <code>mobster</code> we provide the implementation of the model’s density function (<code>ddbpmm</code>, density Dirichlet Beta Pareto mixture model), and a sampler (<code>rdbpmm</code>) which is used internally by <code>random_dataset</code> to generate the data.</p>
<div class="sourceCode" id="cb4"><pre class="downlit">
<span class="co"># Data, in the MOBSTER input format with a "VAF" column.</span>
<span class="fu"><a href="https://rdrr.io/r/base/print.html">print</a></span>(<span class="kw">dataset</span><span class="op">$</span><span class="kw">data</span>)
<span class="co">#&gt; # A tibble: 5,000 x 2</span>
<span class="co">#&gt;      VAF cluster</span>
<span class="co">#&gt;    &lt;dbl&gt; &lt;chr&gt;  </span>
<span class="co">#&gt;  1 0.856 C1     </span>
<span class="co">#&gt;  2 0.853 C1     </span>
<span class="co">#&gt;  3 0.844 C1     </span>
<span class="co">#&gt;  4 0.827 C1     </span>
<span class="co">#&gt;  5 0.855 C1     </span>
<span class="co">#&gt;  6 0.854 C1     </span>
<span class="co">#&gt;  7 0.878 C1     </span>
<span class="co">#&gt;  8 0.865 C1     </span>
<span class="co">#&gt;  9 0.845 C1     </span>
<span class="co">#&gt; 10 0.874 C1     </span>
<span class="co">#&gt; # … with 4,990 more rows</span>

<span class="co"># We rename the `cluster` column (that is a reserved name)</span>
<span class="kw">dataset</span><span class="op">$</span><span class="kw">data</span> <span class="op">=</span> <span class="kw">dataset</span><span class="op">$</span><span class="kw">data</span> <span class="op">%&gt;%</span> <span class="kw">dplyr</span>::<span class="fu"><a href="https://dplyr.tidyverse.org/reference/rename.html">rename</a></span>(gen_clusters = <span class="kw">cluster</span>)

<span class="co"># The generated model contains the parameters of the Beta components (a and b),</span>
<span class="co"># the shape and scale of the tail, and the mixing proportion. </span>
<span class="fu"><a href="https://rdrr.io/r/base/print.html">print</a></span>(<span class="kw">dataset</span><span class="op">$</span><span class="kw">model</span>)
<span class="co">#&gt; $a</span>
<span class="co">#&gt;      C1      C2 </span>
<span class="co">#&gt; 72.6286 30.5932 </span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $b</span>
<span class="co">#&gt;       C1       C2 </span>
<span class="co">#&gt; 13.07363 24.93900 </span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $shape</span>
<span class="co">#&gt; [1] 1</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $scale</span>
<span class="co">#&gt; [1] 0.05</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $pi</span>
<span class="co">#&gt;      Tail        C1        C2 </span>
<span class="co">#&gt; 0.3431634 0.3331162 0.3237204</span>

<span class="co"># A plot object (ggplot) is available where each data-point is coloured by </span>
<span class="co"># its generative mixture component. The vertical lines annontate the means of</span>
<span class="co"># the sampled Beta distributions.</span>
<span class="fu"><a href="https://rdrr.io/r/base/print.html">print</a></span>(<span class="kw">dataset</span><span class="op">$</span><span class="kw">plot</span>)
</pre></div>
<p><img src="a1_introduction_files/figure-html/unnamed-chunk-5-1.png" width="384"></p>
</div>
<div id="fitting-a-dataset" class="section level1">
<h1 class="hasAnchor">
<a href="#fitting-a-dataset" class="anchor"></a>Fitting a dataset</h1>
<p>Function <code>mobster_fit</code> fits a MOBSTER model.</p>
<p>The function implements a model-selection routine that by default scores models by their <code>reICL</code> (<em>reduced Integrative Classification Likelihood</em>) score, a variant to the popular <code>BIC</code> that uses the entropy of the latent variables of the mixture. <code>reICL</code> is discussed in the main paper.</p>
<p>This function has several parameters to customize the fitting procedure, and a set of special pre-parametrised runs that can be activated with parameter <code>auto_setup</code>. Here we use <code>auto_setup = "FAST"</code>, an automatic setup for a fast run; its parameters are accessible through an internal package function.</p>
<div class="sourceCode" id="cb5"><pre class="downlit">
<span class="co"># Hidden function (:::)</span>
<span class="kw">mobster</span>:::<span class="fu">template_parameters_fast_setup</span>()
<span class="co">#&gt; $K</span>
<span class="co">#&gt; [1] 1 2</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $samples</span>
<span class="co">#&gt; [1] 2</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $init</span>
<span class="co">#&gt; [1] "random"</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $tail</span>
<span class="co">#&gt; [1]  TRUE FALSE</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $epsilon</span>
<span class="co">#&gt; [1] 1e-06</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $maxIter</span>
<span class="co">#&gt; [1] 100</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $fit.type</span>
<span class="co">#&gt; [1] "MM"</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $seed</span>
<span class="co">#&gt; [1] 12345</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $model.selection</span>
<span class="co">#&gt; [1] "reICL"</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $trace</span>
<span class="co">#&gt; [1] FALSE</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $parallel</span>
<span class="co">#&gt; [1] FALSE</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $pi_cutoff</span>
<span class="co">#&gt; [1] 0.02</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $N_cutoff</span>
<span class="co">#&gt; [1] 10</span>
</pre></div>
<p>Compared to these, default parameters test more extensive types of fits (i.e., more clones, longer fits, higher number of replicates etc.). We usually use the fast parametrisation to obtain a first fit of the data and, if not satisfied, we run customised calls of <code>mobster_fit</code>.</p>
<div class="sourceCode" id="cb6"><pre class="downlit">
<span class="co"># Fast run with auto_setup = "FAST"</span>
<span class="kw">fit</span> <span class="op">=</span> <span class="fu"><a href="../reference/mobster_fit.html">mobster_fit</a></span>(
  <span class="kw">dataset</span><span class="op">$</span><span class="kw">data</span>,    
  auto_setup = <span class="st">"FAST"</span>
  )
<span class="co">#&gt;  [ MOBSTER fit ]</span>
<span class="co">#&gt; ✔ Loaded input data, n = 5000.</span>
<span class="co">#&gt; ❯ n = 5000. Mixture with k = 1,2 Beta(s). Pareto tail: TRUE and FALSE. Output clusters with π &gt; 0.02</span>
<span class="co">#&gt; and n &gt; 10.</span>
<span class="co">#&gt; ! mobster automatic setup FAST for the analysis.</span>
<span class="co">#&gt; ❯ Scoring (without parallel) 2 x 2 x 2 = 8 models by reICL.</span>
<span class="co">#&gt; ℹ MOBSTER fits completed in 21.1s.</span>
<span class="co">#&gt; ── [ MOBSTER ] My MOBSTER model n = 5000 with k = 2 Beta(s) and a tail ───────────────────────────────</span>
<span class="co">#&gt; ● Clusters: π = 35% [C2], 33% [Tail], and 32% [C1], with π &gt; 0.</span>
<span class="co">#&gt; ● Tail [n = 1567, 33%] with alpha = 1.3.</span>
<span class="co">#&gt; ● Beta C1 [n = 1636, 32%] with mean = 0.85.</span>
<span class="co">#&gt; ● Beta C2 [n = 1797, 35%] with mean = 0.55.</span>
<span class="co">#&gt; ℹ Score(s): NLL = -2558.66; ICL = -4657.61 (-5009.64), H = 383.05 (31.02). Fit converged by MM in 28</span>
<span class="co">#&gt; steps.</span>
</pre></div>
<p>A call of <code>mobster_fit</code> will return a list with 3 elements:</p>
<ul>
<li>the best fit <code>fit$best</code>, according to the selected scoring method;</li>
<li>
<code>fit$runs</code>, a list with the ranked fits; <code>best</code> matches the head of this list;</li>
<li>
<code>fit$fits.table</code>, a table that summarises the scores for each one of the runs.</li>
</ul>
<p>Each fit object (<code>best</code> or any object stored in <code>runs</code>) is from the S3 class <code>dbpmm</code>.</p>
<div class="sourceCode" id="cb7"><pre class="downlit">
<span class="co"># Print the best model</span>
<span class="fu"><a href="https://rdrr.io/r/base/print.html">print</a></span>(<span class="kw">fit</span><span class="op">$</span><span class="kw">best</span>)
<span class="co">#&gt; ── [ MOBSTER ] My MOBSTER model n = 5000 with k = 2 Beta(s) and a tail ───────────────────────────────</span>
<span class="co">#&gt; ● Clusters: π = 35% [C2], 33% [Tail], and 32% [C1], with π &gt; 0.</span>
<span class="co">#&gt; ● Tail [n = 1567, 33%] with alpha = 1.3.</span>
<span class="co">#&gt; ● Beta C1 [n = 1636, 32%] with mean = 0.85.</span>
<span class="co">#&gt; ● Beta C2 [n = 1797, 35%] with mean = 0.55.</span>
<span class="co">#&gt; ℹ Score(s): NLL = -2558.66; ICL = -4657.61 (-5009.64), H = 383.05 (31.02). Fit converged by MM in 28</span>
<span class="co">#&gt; steps.</span>

<span class="co"># Print top-3 models</span>
<span class="fu"><a href="https://rdrr.io/r/base/print.html">print</a></span>(<span class="kw">fit</span><span class="op">$</span><span class="kw">runs</span>[[<span class="fl">1</span>]]) 
<span class="co">#&gt; ── [ MOBSTER ] My MOBSTER model n = 5000 with k = 2 Beta(s) and a tail ───────────────────────────────</span>
<span class="co">#&gt; ● Clusters: π = 35% [C2], 33% [Tail], and 32% [C1], with π &gt; 0.</span>
<span class="co">#&gt; ● Tail [n = 1567, 33%] with alpha = 1.3.</span>
<span class="co">#&gt; ● Beta C1 [n = 1636, 32%] with mean = 0.85.</span>
<span class="co">#&gt; ● Beta C2 [n = 1797, 35%] with mean = 0.55.</span>
<span class="co">#&gt; ℹ Score(s): NLL = -2558.66; ICL = -4657.61 (-5009.64), H = 383.05 (31.02). Fit converged by MM in 28</span>
<span class="co">#&gt; steps.</span>
<span class="fu"><a href="https://rdrr.io/r/base/print.html">print</a></span>(<span class="kw">fit</span><span class="op">$</span><span class="kw">runs</span>[[<span class="fl">2</span>]])
<span class="co">#&gt; ── [ MOBSTER ] My MOBSTER model n = 5000 with k = 1 Beta(s) and a tail ───────────────────────────────</span>
<span class="co">#&gt; ● Clusters: π = 71% [C1] and 29% [Tail], with π &gt; 0.</span>
<span class="co">#&gt; ● Tail [n = 1429, 29%] with alpha = 1.5.</span>
<span class="co">#&gt; ● Beta C1 [n = 3571, 71%] with mean = 0.68.</span>
<span class="co">#&gt; ℹ Score(s): NLL = -1446.64; ICL = -2379.16 (-2842.18), H = 463.01 (0). Fit converged by MM in 22</span>
<span class="co">#&gt; steps.</span>
<span class="fu"><a href="https://rdrr.io/r/base/print.html">print</a></span>(<span class="kw">fit</span><span class="op">$</span><span class="kw">runs</span>[[<span class="fl">3</span>]])
<span class="co">#&gt; ── [ MOBSTER ] My MOBSTER model n = 5000 with k = 1 Beta(s) and a tail ───────────────────────────────</span>
<span class="co">#&gt; ● Clusters: π = 71% [C1] and 29% [Tail], with π &gt; 0.</span>
<span class="co">#&gt; ● Tail [n = 1429, 29%] with alpha = 1.5.</span>
<span class="co">#&gt; ● Beta C1 [n = 3571, 71%] with mean = 0.68.</span>
<span class="co">#&gt; ℹ Score(s): NLL = -1446.64; ICL = -2379.17 (-2842.18), H = 463.01 (0). Fit converged by MM in 22</span>
<span class="co">#&gt; steps.</span>
</pre></div>
<p>Usually, one keeps working with the <code>best</code> model fit. From that it is possible to extract the results of the fit, and the clustering assignments. The output is a copy of the input data, with a column reporting the model’s latent variables (LVs) and the <code>cluster</code> assignment (<em>hard clustering</em>).</p>
<div class="sourceCode" id="cb8"><pre class="downlit">
<span class="co"># All assignments</span>
<span class="fu"><a href="../reference/Clusters.html">Clusters</a></span>(<span class="kw">fit</span><span class="op">$</span><span class="kw">best</span>)
<span class="co">#&gt; # A tibble: 5,000 x 6</span>
<span class="co">#&gt;      VAF gen_clusters cluster    Tail    C1           C2</span>
<span class="co">#&gt;    &lt;dbl&gt; &lt;chr&gt;        &lt;chr&gt;     &lt;dbl&gt; &lt;dbl&gt;        &lt;dbl&gt;</span>
<span class="co">#&gt;  1 0.856 C1           C1      0.00371 0.996 0.000000547 </span>
<span class="co">#&gt;  2 0.853 C1           C1      0.00375 0.996 0.000000759 </span>
<span class="co">#&gt;  3 0.844 C1           C1      0.00403 0.996 0.00000242  </span>
<span class="co">#&gt;  4 0.827 C1           C1      0.00520 0.995 0.0000187   </span>
<span class="co">#&gt;  5 0.855 C1           C1      0.00371 0.996 0.000000571 </span>
<span class="co">#&gt;  6 0.854 C1           C1      0.00373 0.996 0.000000684 </span>
<span class="co">#&gt;  7 0.878 C1           C1      0.00415 0.996 0.0000000246</span>
<span class="co">#&gt;  8 0.865 C1           C1      0.00370 0.996 0.000000171 </span>
<span class="co">#&gt;  9 0.845 C1           C1      0.00398 0.996 0.00000208  </span>
<span class="co">#&gt; 10 0.874 C1           C1      0.00392 0.996 0.0000000480</span>
<span class="co">#&gt; # … with 4,990 more rows</span>

<span class="co"># Assignments with LVs probability above 85%</span>
<span class="fu"><a href="../reference/Clusters.html">Clusters</a></span>(<span class="kw">fit</span><span class="op">$</span><span class="kw">best</span>, cutoff_assignment = <span class="fl">0.85</span>)
<span class="co">#&gt; # A tibble: 5,000 x 6</span>
<span class="co">#&gt;      VAF gen_clusters cluster    Tail    C1           C2</span>
<span class="co">#&gt;    &lt;dbl&gt; &lt;chr&gt;        &lt;chr&gt;     &lt;dbl&gt; &lt;dbl&gt;        &lt;dbl&gt;</span>
<span class="co">#&gt;  1 0.856 C1           C1      0.00371 0.996 0.000000547 </span>
<span class="co">#&gt;  2 0.853 C1           C1      0.00375 0.996 0.000000759 </span>
<span class="co">#&gt;  3 0.844 C1           C1      0.00403 0.996 0.00000242  </span>
<span class="co">#&gt;  4 0.827 C1           C1      0.00520 0.995 0.0000187   </span>
<span class="co">#&gt;  5 0.855 C1           C1      0.00371 0.996 0.000000571 </span>
<span class="co">#&gt;  6 0.854 C1           C1      0.00373 0.996 0.000000684 </span>
<span class="co">#&gt;  7 0.878 C1           C1      0.00415 0.996 0.0000000246</span>
<span class="co">#&gt;  8 0.865 C1           C1      0.00370 0.996 0.000000171 </span>
<span class="co">#&gt;  9 0.845 C1           C1      0.00398 0.996 0.00000208  </span>
<span class="co">#&gt; 10 0.874 C1           C1      0.00392 0.996 0.0000000480</span>
<span class="co">#&gt; # … with 4,990 more rows</span>
</pre></div>
<p>The second call imposes a cut to the assignments with less than 85% probability mass in the LVs.</p>
<p>If you want to assign some new data to the fit model you can use function <code>Clusters_denovo</code>.</p>
</div>
<div id="basic-plots-of-a-fit" class="section level1">
<h1 class="hasAnchor">
<a href="#basic-plots-of-a-fit" class="anchor"></a>Basic plots of a fit</h1>
<p>Clusters can be plot as an histogram with the model density (total and per mixture). By default, <code>mobster</code> names Beta clusters <code>C1</code>, <code>C2</code>, etc. according to the decreasing order of their mean; so <code>C1</code> is always the cluster with highest Beta mean, etc. If the data are diploid mutations, <code>C1</code> should represent clonal mutations.</p>
<div class="sourceCode" id="cb9"><pre class="downlit">
<span class="co"># Plot the best model</span>
<span class="fu"><a href="https://rdrr.io/r/graphics/plot.default.html">plot</a></span>(<span class="kw">fit</span><span class="op">$</span><span class="kw">best</span>)
</pre></div>
<p><img src="a1_introduction_files/figure-html/unnamed-chunk-10-1.png" width="480"></p>
<p>A comparative plot between the fit and data is assembled using <a href="https://cran.r-project.org/web/packages/cowplot/vignettes/introduction.html">cowplot</a>.</p>
<div class="sourceCode" id="cb10"><pre class="downlit">
<span class="kw">cowplot</span>::<span class="fu"><a href="https://rdrr.io/pkg/cowplot/man/plot_grid.html">plot_grid</a></span>(
  <span class="kw">dataset</span><span class="op">$</span><span class="kw">plot</span>, 
  <span class="fu"><a href="https://rdrr.io/r/graphics/plot.default.html">plot</a></span>(<span class="kw">fit</span><span class="op">$</span><span class="kw">best</span>), 
  ncol = <span class="fl">2</span>,
  align = <span class="st">'h'</span>)
</pre></div>
<p><img src="a1_introduction_files/figure-html/unnamed-chunk-11-1.png" width="960"></p>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p>Developed by <a href="https://sites.google.com/site/giuliocaravagna/">Giulio Caravagna</a>.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.6.0.</p>
</div>

      </footer>
</div>

  


  </body>
</html>
