<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>5. dN/dS statistics • mobster</title>
<!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png">
<link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png">
<link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png">
<link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png">
<link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png">
<link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png">
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="5. dN/dS statistics">
<meta property="og:description" content="mobster">
<meta property="og:image" content="caravagn.github.io/mobster/logo.png">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">mobster</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.1.1</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../index.html">
    <span class="fas fa fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../articles/mobster.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/a1_introduction.html">1. Introduction</a>
    </li>
    <li>
      <a href="../articles/a2_plotting.html">2. Plotting fits</a>
    </li>
    <li>
      <a href="../articles/a3_bootstrap.html">3. Confidence estimation by the bootstrap</a>
    </li>
    <li>
      <a href="../articles/a4_popgen.html">4. Population Genetics statistics</a>
    </li>
    <li>
      <a href="../articles/a5_dnds.html">5. dN/dS statistics</a>
    </li>
    <li>
      <a href="../articles/a6_CloneTrees.html">6. Clone trees from clusters</a>
    </li>
    <li>
      <a href="../articles/a7_Example_tumour_simulation.html">7. Visualizing subclonal expansions</a>
    </li>
  </ul>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/caravagn/mobster/">
    <span class="fab fa fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>5. dN/dS statistics</h1>
                        <h4 class="author">Giulio Caravagna</h4>
            
            <h4 class="date">13 September, 2020</h4>
      
      <small class="dont-index">Source: <a href="https://github.com/caravagn/mobstervignettes/a5_dnds.Rmd"><code>vignettes/a5_dnds.Rmd</code></a></small>
      <div class="hidden name"><code>a5_dnds.Rmd</code></div>

    </div>

    
    
<div class="sourceCode" id="cb1"><pre class="downlit">
<span class="fu"><a href="https://rdrr.io/r/base/library.html">library</a></span>(<span class="kw"><a href="https://github.com/caravagn/mobster">mobster</a></span>)
<span class="fu"><a href="https://rdrr.io/r/base/library.html">library</a></span>(<span class="kw"><a href="https://tidyr.tidyverse.org">tidyr</a></span>)
<span class="fu"><a href="https://rdrr.io/r/base/library.html">library</a></span>(<span class="kw"><a href="https://dplyr.tidyverse.org">dplyr</a></span>)
</pre></div>
<div id="computing-dnds-values" class="section level1">
<h1 class="hasAnchor">
<a href="#computing-dnds-values" class="anchor"></a>Computing dnds values</h1>
<p><code>mobster</code> interfaces with the <a href="https://github.com/im3sanger/dndscv/">dndscv</a> R package to compute dN/dS values from its output clusters. The method implemented in <code>dndscv</code> is described in <em>Martincorena, et al.</em> <em>“Universal patterns of selection in cancer and somatic tissues”</em>, Cell 171.5 (2017): 1029-1041; <a href="https://www.ncbi.nlm.nih.gov/pubmed/29056346">PMID 29056346</a>).</p>
<p><strong>Requirements.</strong> In order to be able to compute dN/dS values mutations data must store their <em>genomic coordinates</em>:</p>
<ul>
<li>chromosome location <code>chrom</code>,</li>
<li>position <code>from</code>,</li>
<li>reference alleles <code>alt</code> and <code>ref</code>.</li>
</ul>
<p>Besides, it is important to know what is the reference genome used to align the genome; this information will be used by <code>dndscv</code> to annotate input mutations.</p>
<p>We show this analysis with the fits for one of the lung samples available in the package.</p>
<div class="sourceCode" id="cb2"><pre class="downlit">
<span class="kw">fit</span> <span class="op">=</span> <span class="kw">mobster</span>::<span class="kw"><a href="../reference/LUFF76_lung_sample.html">LUFF76_lung_sample</a></span>

<span class="co"># Print and plot the model</span>
<span class="fu"><a href="https://rdrr.io/r/base/print.html">print</a></span>(<span class="kw">fit</span><span class="op">$</span><span class="kw">best</span>)
<span class="co">#&gt; ── [ MOBSTER ]  n = 2298 with k = 1 Beta(s) and a tail ───────────────────────────────────────────────</span>
<span class="co">#&gt; ● Clusters: π = 50% [Tail] and 50% [C1], with π &gt; 0.</span>
<span class="co">#&gt; ● Tail [n = 1076, 50%] with alpha = 2.</span>
<span class="co">#&gt; ● Beta C1 [n = 1222, 50%] with mean = 0.25.</span>
<span class="co">#&gt; ℹ Score(s): NLL = -3095.24; ICL = -5784.22 (-6144.04), H = 359.82 (0). Fit converged by MM in 51</span>
<span class="co">#&gt; steps.</span>
<span class="fu"><a href="https://rdrr.io/r/graphics/plot.default.html">plot</a></span>(<span class="kw">fit</span><span class="op">$</span><span class="kw">best</span>)
</pre></div>
<p><img src="a5_dnds_files/figure-html/unnamed-chunk-3-1.png" width="480"></p>
<p>We compute the values using the clustering assignments from the best fit.</p>
<div class="sourceCode" id="cb3"><pre class="downlit">
<span class="kw">clusters</span> <span class="op">=</span> <span class="fu"><a href="../reference/Clusters.html">Clusters</a></span>(<span class="kw">fit</span><span class="op">$</span><span class="kw">best</span>)
<span class="fu"><a href="https://rdrr.io/r/base/print.html">print</a></span>(<span class="kw">clusters</span>)
<span class="co">#&gt; # A tibble: 2,298 x 14</span>
<span class="co">#&gt;      Key Callers t_alt_count t_ref_count Variant_Classif…    DP    VAF chr  </span>
<span class="co">#&gt;    &lt;int&gt; &lt;chr&gt;         &lt;int&gt;       &lt;int&gt; &lt;chr&gt;            &lt;int&gt;  &lt;dbl&gt; &lt;chr&gt;</span>
<span class="co">#&gt;  1   330 mutect…          23         149 intergenic         172 0.134  chr2 </span>
<span class="co">#&gt;  2   331 mutect…          53         133 intergenic         186 0.285  chr2 </span>
<span class="co">#&gt;  3   332 mutect…           8         132 intergenic         140 0.0571 chr2 </span>
<span class="co">#&gt;  4   334 mutect…          26         112 intergenic         138 0.188  chr2 </span>
<span class="co">#&gt;  5   335 mutect…          40         102 intronic           142 0.282  chr2 </span>
<span class="co">#&gt;  6   336 mutect…          32          95 intronic           127 0.252  chr2 </span>
<span class="co">#&gt;  7   337 mutect…          28         134 intergenic         162 0.173  chr2 </span>
<span class="co">#&gt;  8   338 mutect…          14         148 intergenic         162 0.0864 chr2 </span>
<span class="co">#&gt;  9   339 mutect…          11         108 intronic           119 0.0924 chr2 </span>
<span class="co">#&gt; 10   340 mutect…          11         171 intergenic         182 0.0604 chr2 </span>
<span class="co">#&gt; # … with 2,288 more rows, and 6 more variables: from &lt;chr&gt;, ref &lt;chr&gt;,</span>
<span class="co">#&gt; #   alt &lt;chr&gt;, cluster &lt;chr&gt;, Tail &lt;dbl&gt;, C1 &lt;dbl&gt;</span>
</pre></div>
<p>The available clusters are <code>C1</code> and <code>Tail</code>; <code>C1</code> is the clonal cluster. We compute dN/dS with the default parameters.</p>
<div class="sourceCode" id="cb4"><pre class="downlit">
<span class="co"># Run by cluster and default gene list</span>
<span class="kw">dnds_stats</span> <span class="op">=</span> <span class="fu"><a href="../reference/dnds.html">dnds</a></span>(
  <span class="kw">clusters</span>,
  gene_list = <span class="kw">NULL</span>
)
<span class="co">#&gt; Missing 'sample' column, assuming mutations from a single patient (adding a sample label otherwise).</span>
<span class="co">#&gt; ℹ 2298 mutations; 'by cluster' groups in 0 samples, with no genes (default dndscv).</span>
<span class="co">#&gt; [refdb = hg19] Removing chr from chromosome names for hg19 reference compatability</span>
<span class="co">#&gt; </span>
<span class="co">#&gt;   C1 Tail </span>
<span class="co">#&gt; 1222 1076</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; ── Running dndscv ────────────────────────────────────────────────────────────────────────────────────</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; ── Group Tail</span>
<span class="co">#&gt; [1] Loading the environment...</span>
<span class="co">#&gt; [2] Annotating the mutations...</span>
<span class="co">#&gt; [3] Estimating global rates...</span>
<span class="co">#&gt; [4] Running dNdSloc...</span>
<span class="co">#&gt; [5] Running dNdScv...</span>
<span class="co">#&gt;     Regression model for substitutions (theta = 6.69e-05).</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; ── Group C1</span>
<span class="co">#&gt; [1] Loading the environment...</span>
<span class="co">#&gt; [2] Annotating the mutations...</span>
<span class="co">#&gt; [3] Estimating global rates...</span>
<span class="co">#&gt; [4] Running dNdSloc...</span>
<span class="co">#&gt; [5] Running dNdScv...</span>
<span class="co">#&gt; dndscv error</span>
<span class="co">#&gt; Error in while ((it &lt;- it + 1) &lt; limit &amp;&amp; abs(del) &gt; eps) {: missing value where TRUE/FALSE needed</span>
<span class="co">#&gt; ── dndscv results ──────────────────────────────────────────────────── wall, wmis, wnon, wspl, wtru ──</span>
<span class="co">#&gt; # A tibble: 5 x 5</span>
<span class="co">#&gt;   name            mle cilow cihigh dnds_group</span>
<span class="co">#&gt;   &lt;chr&gt;         &lt;dbl&gt; &lt;dbl&gt;  &lt;dbl&gt; &lt;chr&gt;     </span>
<span class="co">#&gt; 1 wmis  0.731         0.113   4.73 Tail      </span>
<span class="co">#&gt; 2 wnon  0.00000000804 0     Inf    Tail      </span>
<span class="co">#&gt; 3 wspl  0.0000000260  0     Inf    Tail      </span>
<span class="co">#&gt; 4 wtru  0.0000000170  0     Inf    Tail      </span>
<span class="co">#&gt; 5 wall  0.701         0.109   4.51 Tail</span>
</pre></div>
<p>The statistics can be computed for a custom grouping of the clusters. Here it does not make much difference because we have only the clonal cluster, and the tail; but if we had one subclone <code>C2</code> we could have pooled together the mutations in the clones using</p>
<div class="sourceCode" id="cb5"><pre class="downlit">
<span class="co"># Not run here</span>
<span class="kw">dnds_stats</span> <span class="op">=</span> <span class="fu"><a href="../reference/dnds.html">dnds</a></span>(
  <span class="kw">clusters</span>,
  mapping = <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span>(`C1` = <span class="st">'Non-tail'</span>, `C2` = <span class="st">'Non-tail'</span>, `Tail` = <span class="st">'Tail'</span>),
  gene_list = <span class="kw">NULL</span>
)
</pre></div>
<p>In the above analysis we have run <code>dndscv</code> using the default gene list (<code>gene_list = NULL</code>). Notice that errors raised by <code>dndscv</code> are intercepted by <code>mobster</code>; some of this errors might originate from a dataset with not enough substitutions to compute dN/dS.</p>
<p>The call returns:</p>
<ul>
<li>the table computed by <code>dndscv</code>, where column <code>dnds_group</code> labels the group.</li>
<li>a <code>ggplot</code> plot of the point estimates and the confidence interval;</li>
</ul>
<div class="sourceCode" id="cb6"><pre class="downlit">
<span class="co"># Summary statistics</span>
<span class="fu"><a href="https://rdrr.io/r/base/print.html">print</a></span>(<span class="kw">dnds_stats</span><span class="op">$</span><span class="kw">dnds_summary</span>)
<span class="co">#&gt; # A tibble: 5 x 5</span>
<span class="co">#&gt;   name            mle cilow cihigh dnds_group</span>
<span class="co">#&gt;   &lt;chr&gt;         &lt;dbl&gt; &lt;dbl&gt;  &lt;dbl&gt; &lt;chr&gt;     </span>
<span class="co">#&gt; 1 wmis  0.731         0.113   4.73 Tail      </span>
<span class="co">#&gt; 2 wnon  0.00000000804 0     Inf    Tail      </span>
<span class="co">#&gt; 3 wspl  0.0000000260  0     Inf    Tail      </span>
<span class="co">#&gt; 4 wtru  0.0000000170  0     Inf    Tail      </span>
<span class="co">#&gt; 5 wall  0.701         0.109   4.51 Tail</span>

<span class="co"># Table observation countns</span>
<span class="fu"><a href="https://rdrr.io/r/base/print.html">print</a></span>(<span class="kw">dnds_stats</span><span class="op">$</span><span class="kw">dndscv_table</span>)
<span class="co">#&gt; # A tibble: 20,091 x 15</span>
<span class="co">#&gt;    gene_name n_syn n_mis n_non n_spl wmis_cv wnon_cv wspl_cv pmis_cv ptrunc_cv</span>
<span class="co">#&gt;    &lt;chr&gt;     &lt;dbl&gt; &lt;dbl&gt; &lt;dbl&gt; &lt;dbl&gt;   &lt;dbl&gt;   &lt;dbl&gt;   &lt;dbl&gt;   &lt;dbl&gt;     &lt;dbl&gt;</span>
<span class="co">#&gt;  1 ATF4          0     1     0     0   5262.       0       0 1.00e-4     0.998</span>
<span class="co">#&gt;  2 C14orf79      0     1     0     0   4702.       0       0 1.13e-4     0.998</span>
<span class="co">#&gt;  3 FAM189A1      0     1     0     0   3155.       0       0 1.72e-4     0.997</span>
<span class="co">#&gt;  4 RNF150        0     1     0     0   2793.       0       0 1.96e-4     0.998</span>
<span class="co">#&gt;  5 NEO1          0     1     0     0   1141.       0       0 5.09e-4     0.994</span>
<span class="co">#&gt;  6 TTN           0     0     0     0      0        0       0 8.36e-1     0.968</span>
<span class="co">#&gt;  7 MUC16         0     0     0     0      0        0       0 8.95e-1     0.990</span>
<span class="co">#&gt;  8 OBSCN         0     0     0     0      0        0       0 9.15e-1     0.989</span>
<span class="co">#&gt;  9 SYNE1         0     0     0     0      0        0       0 9.17e-1     0.985</span>
<span class="co">#&gt; 10 NEB           0     0     0     0      0        0       0 9.19e-1     0.978</span>
<span class="co">#&gt; # … with 20,081 more rows, and 5 more variables: pallsubs_cv &lt;dbl&gt;,</span>
<span class="co">#&gt; #   qmis_cv &lt;dbl&gt;, qtrunc_cv &lt;dbl&gt;, qallsubs_cv &lt;dbl&gt;, dnds_group &lt;chr&gt;</span>

<span class="co"># Plot</span>
<span class="fu"><a href="https://rdrr.io/r/base/print.html">print</a></span>(<span class="kw">dnds_stats</span><span class="op">$</span><span class="kw">plot</span>)
</pre></div>
<p><img src="a5_dnds_files/figure-html/unnamed-chunk-7-1.png" width="864"></p>
<p>The default plot contains results obtained from all substitution models available in <code>dndscv</code>. Specific models can be required using the parameters of the <code>dnds</code> function.</p>
</div>
<div id="using-custom-genes-lists" class="section level1">
<h1 class="hasAnchor">
<a href="#using-custom-genes-lists" class="anchor"></a>Using custom genes lists</h1>
<p>A custom list of genes can be supplied in the call to <code>dnds</code> as the variable <code>genes_list</code>; the package provides 4 lists of interests for this type of computation:</p>
<ul>
<li>a list of driver genes compiled in <code>Martincorena et al. Cell 171.5 (2017): 1029-1041.</code>;</li>
<li>a list of driver genes compiled in <code>Tarabichi, et al. Nature Genetics 50.12 (2018): 1630.</code>;</li>
<li>a list of essential genes compiled in <code>Wang et al. Science 350.6264 (2015): 1096-1101.</code>;</li>
<li>a list of essential genes compiled in <code>Bloomen et al. Science 350.6264 (2015): 1092-1096.</code>.</li>
</ul>
<p>which are available to load.</p>
<div class="sourceCode" id="cb7"><pre class="downlit">
<span class="co"># Load the list</span>
<span class="fu"><a href="https://rdrr.io/r/utils/data.html">data</a></span>(<span class="st">'cancer_genes_dnds'</span>, package = <span class="st">'mobster'</span>)

<span class="co"># Each sublist is a list </span>
<span class="fu"><a href="https://rdrr.io/r/base/print.html">print</a></span>(<span class="fu"><a href="https://rdrr.io/r/base/lapply.html">lapply</a></span>(<span class="kw">cancer_genes_dnds</span>, <span class="kw">head</span>))
<span class="co">#&gt; $Martincorena_drivers</span>
<span class="co">#&gt; [1] "CCDC6"     "EIF1AX"    "HIST1H2BD" "MED12"     "POLE"      "SMARCB1"  </span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $Tarabichi_drivers</span>
<span class="co">#&gt; [1] "ACVR1"  "ACVR1B" "AKT1"   "ALK"    "AMER1"  "APC"   </span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $Wang_essentials</span>
<span class="co">#&gt; [1] "ABL1"    "RPL23A"  "AARS2"   "TRMT112" "FARSA"   "ABCB7"  </span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $Bloomen_essentials</span>
<span class="co">#&gt; [1] "AARS"     "AASDHPPT" "AATF"     "ABCB7"    "ABCE1"    "ABCF1"</span>
</pre></div>
<p>A custom gene list can be used as follows.</p>
<div class="sourceCode" id="cb8"><pre class="downlit">
<span class="co"># Not run here</span>
<span class="kw">dnds_stats</span> <span class="op">=</span> <span class="fu"><a href="../reference/dnds.html">dnds</a></span>(
  <span class="kw">clusters</span>,
  mapping = <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span>(`C1` = <span class="st">'Non-tail'</span>, `C2` = <span class="st">'Non-tail'</span>, `C3` = <span class="st">'Non-tail'</span>, `Tail` = <span class="st">'Tail'</span>),
  gene_list = <span class="kw">cancer_genes_dnds</span><span class="op">$</span><span class="kw">Martincorena_drivers</span>
)
</pre></div>
</div>
<div id="pooling-data-from-multiple-patients" class="section level1">
<h1 class="hasAnchor">
<a href="#pooling-data-from-multiple-patients" class="anchor"></a>Pooling data from multiple patients</h1>
<p>The input format of the <code>dnds</code> function allows to pool data from several fits at once. We pool data from the 2 datasets available in the package.</p>
<div class="sourceCode" id="cb9"><pre class="downlit">
<span class="co"># 2 lung samples</span>
<span class="fu"><a href="https://rdrr.io/r/utils/data.html">data</a></span>(<span class="st">'LU4_lung_sample'</span>, package = <span class="st">'mobster'</span>)
<span class="fu"><a href="https://rdrr.io/r/utils/data.html">data</a></span>(<span class="st">'LUFF76_lung_sample'</span>, package = <span class="st">'mobster'</span>)
</pre></div>
<p>We pool the data selecting the required columns.</p>
<div class="sourceCode" id="cb10"><pre class="downlit">
<span class="kw">dnds_multi</span> <span class="op">=</span> <span class="fu"><a href="../reference/dnds.html">dnds</a></span>(
  <span class="fu"><a href="https://rdrr.io/r/base/cbind.html">rbind</a></span>(
    <span class="fu"><a href="../reference/Clusters.html">Clusters</a></span>(<span class="kw">LU4_lung_sample</span><span class="op">$</span><span class="kw">best</span>) <span class="op">%&gt;%</span> <span class="fu"><a href="https://dplyr.tidyverse.org/reference/select.html">select</a></span>(<span class="kw">chr</span>, <span class="kw">from</span>, <span class="kw">ref</span>, <span class="kw">alt</span>, <span class="kw">cluster</span>) <span class="op">%&gt;%</span> <span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate.html">mutate</a></span>(sample = <span class="st">'LU4'</span>),
    <span class="fu"><a href="../reference/Clusters.html">Clusters</a></span>(<span class="kw">LUFF76_lung_sample</span><span class="op">$</span><span class="kw">best</span>) <span class="op">%&gt;%</span> <span class="fu"><a href="https://dplyr.tidyverse.org/reference/select.html">select</a></span>(<span class="kw">chr</span>, <span class="kw">from</span>, <span class="kw">ref</span>, <span class="kw">alt</span>, <span class="kw">cluster</span>) <span class="op">%&gt;%</span> <span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate.html">mutate</a></span>(sample = <span class="st">'LUFF76'</span>)
  ),
  mapping = <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span>(`C1` = <span class="st">'Non-tail'</span>,  <span class="co"># Pool together all clonal mutations</span>
              `Tail` = <span class="st">'Tail'</span>     <span class="co"># Pool together all tail mutations),</span>
  )
)
<span class="co">#&gt; ℹ 3580 mutations; 2 groups in 2 samples, with no genes (default dndscv).</span>
<span class="co">#&gt; [refdb = hg19] Removing chr from chromosome names for hg19 reference compatability</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; Non-tail     Tail </span>
<span class="co">#&gt;     2194     1386</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; ── Running dndscv ────────────────────────────────────────────────────────────────────────────────────</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; ── Group Non-tail</span>
<span class="co">#&gt; [1] Loading the environment...</span>
<span class="co">#&gt; [2] Annotating the mutations...</span>
<span class="co">#&gt; [3] Estimating global rates...</span>
<span class="co">#&gt; [4] Running dNdSloc...</span>
<span class="co">#&gt; [5] Running dNdScv...</span>
<span class="co">#&gt;     Regression model for substitutions (theta = 17.5).</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; ── Group Tail</span>
<span class="co">#&gt; [1] Loading the environment...</span>
<span class="co">#&gt; [2] Annotating the mutations...</span>
<span class="co">#&gt; [3] Estimating global rates...</span>
<span class="co">#&gt; [4] Running dNdSloc...</span>
<span class="co">#&gt; [5] Running dNdScv...</span>
<span class="co">#&gt;     Regression model for substitutions (theta = 8.54).</span>
<span class="co">#&gt; ── dndscv results ──────────────────────────────────────────────────── wall, wmis, wnon, wspl, wtru ──</span>
<span class="co">#&gt; # A tibble: 10 x 5</span>
<span class="co">#&gt;    name            mle  cilow cihigh dnds_group</span>
<span class="co">#&gt;    &lt;chr&gt;         &lt;dbl&gt;  &lt;dbl&gt;  &lt;dbl&gt; &lt;chr&gt;     </span>
<span class="co">#&gt;  1 wmis  1.16          0.375    3.58 Non-tail  </span>
<span class="co">#&gt;  2 wnon  2.25          0.223   22.8  Non-tail  </span>
<span class="co">#&gt;  3 wspl  0.0000000301  0      Inf    Non-tail  </span>
<span class="co">#&gt;  4 wtru  0.658         0.0716   6.05 Non-tail  </span>
<span class="co">#&gt;  5 wall  1.11          0.370    3.33 Non-tail  </span>
<span class="co">#&gt;  6 wmis  0.508         0.121    2.13 Tail      </span>
<span class="co">#&gt;  7 wnon  0.00000000835 0      Inf    Tail      </span>
<span class="co">#&gt;  8 wspl  0.0000000461  0      Inf    Tail      </span>
<span class="co">#&gt;  9 wtru  0.0000000201  0      Inf    Tail      </span>
<span class="co">#&gt; 10 wall  0.472         0.115    1.95 Tail</span>

<span class="fu"><a href="https://rdrr.io/r/base/print.html">print</a></span>(<span class="kw">dnds_multi</span><span class="op">$</span><span class="kw">plot</span>)
</pre></div>
<p><img src="a5_dnds_files/figure-html/unnamed-chunk-11-1.png" width="864"></p>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p>Developed by <a href="https://sites.google.com/site/giuliocaravagna/">Giulio Caravagna</a>.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.6.0.</p>
</div>

      </footer>
</div>

  


  </body>
</html>
