<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Evolutionary analyses of mobster fits • mobster</title>
<!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png">
<link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png">
<link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png">
<link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png">
<link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png">
<link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png">
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.3.1/jquery.min.js" integrity="sha256-FgpCb/KJQlLNfOu91ta32o/NMZxltwRo8QtmkMRdAu8=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.3.7/css/bootstrap.min.css" integrity="sha256-916EbMg70RQy9LHiGkXzG8hSg9EdNy97GazNG/aiY1w=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.3.7/js/bootstrap.min.js" integrity="sha256-U5ZEeKfGNOja007MMD3YBI0A3OSZOQbeG6z2f2Y0hu8=" crossorigin="anonymous"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.7.1/css/all.min.css" integrity="sha256-nAmazAk6vS34Xqo0BSrTb+abbtFlgsFK7NKSi6o7Y78=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.7.1/css/v4-shims.min.css" integrity="sha256-6qHlizsOWFskGlwVOKuns+D1nB6ssZrHQrNj1wGplHc=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.4/clipboard.min.js" integrity="sha256-FiZwavyI2V6+EXO1U+xzLG3IKldpiTFf3153ea9zikQ=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.9.4/headroom.min.js" integrity="sha256-DJFC1kqIhelURkuza0AvYal5RxMtpzLjFhsnVIeuk+U=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.9.4/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="Evolutionary analyses of mobster fits">
<meta property="og:description" content="">
<meta property="og:image" content="/logo.png">
<meta name="twitter:card" content="summary">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body>
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">mobster</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.1.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../index.html">
    <span class="fas fa fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../articles/mobster.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/bootstrap.html">Confidence estimation via the bootstrap</a>
    </li>
    <li>
      <a href="../articles/evolutionary_analysis.html">Evolutionary analyses of mobster fits</a>
    </li>
    <li>
      <a href="../articles/introduction.html">Introduction</a>
    </li>
    <li>
      <a href="../articles/plotting.html">Plotting mobster fits</a>
    </li>
  </ul>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right"></ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1>Evolutionary analyses of mobster fits</h1>
                        <h4 class="author">Giulio Caravagna</h4>
            
            <h4 class="date">September 2019</h4>
      
      
      <div class="hidden name"><code>evolutionary_analysis.Rmd</code></div>

    </div>

    
    
<p>This vignette describes the evolutionary analyses that are available in <a href="https://github.com/caravagn/MOBSTER">MOBSTER</a>, and include:</p>
<ul>
<li>computation of dnds values via the package <code>dndscv</code>;</li>
<li>computation of evolutionary parameters;</li>
</ul>
<div class="sourceCode" id="cb1"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb1-1" data-line-number="1"><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span>(mobster)</a></code></pre></div>
<div id="computing-dnds-values" class="section level2">
<h2 class="hasAnchor">
<a href="#computing-dnds-values" class="anchor"></a>Computing dnds values</h2>
<p>Dn/Ds values are computed wrapping a call to the <a href="https://github.com/im3sanger/dndscv/">dndscv</a> R package, developed by Martincorena at Sanger.</p>
<blockquote>
<p>Martincorena, et al. “Universal patterns of selection in cancer and somatic tissues.” Cell 171.5 (2017): 1029-1041.</p>
</blockquote>
<p>These values can only be computed if the available mutation calls have the following information:</p>
<ul>
<li>the chromosome location, labelled as <code>chrom</code>;</li>
<li>the mutation position, labelled as <code>from</code>;</li>
<li>the mutant and the reference alleles, labelled as <code>alt</code> and <code>ref</code>.</li>
</ul>
<p>To show these analysis we use the fits for the <code>PD4120a</code> breast cancer sample available in MOBSTER.</p>
<div class="sourceCode" id="cb2"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb2-1" data-line-number="1"><span class="kw"><a href="https://rdrr.io/r/utils/data.html">data</a></span>(<span class="st">'PD4120a_breast_sample'</span>, <span class="dt">package =</span> <span class="st">'mobster'</span>)</a>
<a class="sourceLine" id="cb2-2" data-line-number="2"></a>
<a class="sourceLine" id="cb2-3" data-line-number="3"><span class="co"># Print and plot the model</span></a>
<a class="sourceLine" id="cb2-4" data-line-number="4"><span class="kw"><a href="https://rdrr.io/r/base/print.html">print</a></span>(PD4120a_breast_sample<span class="op">$</span>best)</a>
<a class="sourceLine" id="cb2-5" data-line-number="5"><span class="co">#&gt; ── [ MOBSTER ]  n = 4643 with k = 3 Beta(s) and a tail ────────────────────────────────────────────────────────────────</span></a>
<a class="sourceLine" id="cb2-6" data-line-number="6"><span class="co">#&gt; ● Clusters: π = 37% [C2], 26% [Tail], 24% [C1] and 13% [C3], with π &gt; 0.</span></a>
<a class="sourceLine" id="cb2-7" data-line-number="7"><span class="co">#&gt; ● Tail [n = 1006, 26%] with alpha = 1.5.</span></a>
<a class="sourceLine" id="cb2-8" data-line-number="8"><span class="co">#&gt; ● Beta C1 [n = 1740, 37%] with mean = 0.52.</span></a>
<a class="sourceLine" id="cb2-9" data-line-number="9"><span class="co">#&gt; ● Beta C2 [n = 1167, 24%] with mean = 0.29.</span></a>
<a class="sourceLine" id="cb2-10" data-line-number="10"><span class="co">#&gt; ● Beta C3 [n = 730, 13%] with mean = 0.16.</span></a>
<a class="sourceLine" id="cb2-11" data-line-number="11"><span class="co">#&gt; ℹ Score(s): NLL = -3201.73; ICL = -5346.89 (-6113.83), H = 955.25 (188.31). Fit converged by MM in 259 steps.</span></a>
<a class="sourceLine" id="cb2-12" data-line-number="12"><span class="kw"><a href="https://rdrr.io/r/graphics/plot.html">plot</a></span>(PD4120a_breast_sample<span class="op">$</span>best)</a></code></pre></div>
<p><img src="evolutionary_analysis_files/figure-html/unnamed-chunk-3-1.png" width="480"></p>
<p>To compute the values one first extracts the clustering assignments from the best fit.</p>
<div class="sourceCode" id="cb3"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb3-1" data-line-number="1">clusters =<span class="st"> </span><span class="kw"><a href="../reference/Clusters.html">Clusters</a></span>(PD4120a_breast_sample<span class="op">$</span>best)</a>
<a class="sourceLine" id="cb3-2" data-line-number="2">pio<span class="op">::</span><span class="kw"><a href="https://rdrr.io/pkg/pio/man/pioDisp.html">pioDisp</a></span>(clusters)</a>
<a class="sourceLine" id="cb3-3" data-line-number="3"><span class="co">#&gt; # A tibble: 4,643 x 19</span></a>
<a class="sourceLine" id="cb3-4" data-line-number="4"><span class="co">#&gt;    region gene  chr     from     to ref   alt   vartype VAF.tumour VAF.normal</span></a>
<a class="sourceLine" id="cb3-5" data-line-number="5"><span class="co">#&gt;    &lt;chr&gt;  &lt;chr&gt; &lt;chr&gt;  &lt;int&gt;  &lt;int&gt; &lt;chr&gt; &lt;chr&gt; &lt;chr&gt;        &lt;dbl&gt;      &lt;dbl&gt;</span></a>
<a class="sourceLine" id="cb3-6" data-line-number="6"><span class="co">#&gt;  1 inter… LOC1… chr3   74574  74574 G     A     SNP          0.371          0</span></a>
<a class="sourceLine" id="cb3-7" data-line-number="7"><span class="co">#&gt;  2 inter… LOC1… chr3   80489  80489 C     T     SNP          0.276          0</span></a>
<a class="sourceLine" id="cb3-8" data-line-number="8"><span class="co">#&gt;  3 inter… LOC1… chr3  139091 139091 G     C     SNP          0.049          0</span></a>
<a class="sourceLine" id="cb3-9" data-line-number="9"><span class="co">#&gt;  4 inter… LOC1… chr3  150557 150557 G     C     SNP          0.077          0</span></a>
<a class="sourceLine" id="cb3-10" data-line-number="10"><span class="co">#&gt;  5 inter… LOC1… chr3  190860 190860 T     A     SNP          0.033          0</span></a>
<a class="sourceLine" id="cb3-11" data-line-number="11"><span class="co">#&gt;  6 inter… LOC1… chr3  198896 198896 C     G     SNP          0.347          0</span></a>
<a class="sourceLine" id="cb3-12" data-line-number="12"><span class="co">#&gt;  7 inter… LOC1… chr3  234298 234298 C     T     SNP          0.233          0</span></a>
<a class="sourceLine" id="cb3-13" data-line-number="13"><span class="co">#&gt;  8 intro… CHL1  chr3  271958 271958 C     G     SNP          0.282          0</span></a>
<a class="sourceLine" id="cb3-14" data-line-number="14"><span class="co">#&gt;  9 intro… CHL1  chr3  312028 312028 C     T     SNP          0.083          0</span></a>
<a class="sourceLine" id="cb3-15" data-line-number="15"><span class="co">#&gt; 10 intro… CHL1  chr3  321624 321624 C     T     SNP          0.162          0</span></a>
<a class="sourceLine" id="cb3-16" data-line-number="16"><span class="co">#&gt; # … with 4,633 more rows, and 9 more variables: depth.tumour &lt;dbl&gt;,</span></a>
<a class="sourceLine" id="cb3-17" data-line-number="17"><span class="co">#&gt; #   depth.normal &lt;chr&gt;, NV &lt;dbl&gt;, VAF &lt;dbl&gt;, cluster &lt;chr&gt;, Tail &lt;dbl&gt;,</span></a>
<a class="sourceLine" id="cb3-18" data-line-number="18"><span class="co">#&gt; #   C1 &lt;dbl&gt;, C2 &lt;dbl&gt;, C3 &lt;dbl&gt;</span></a></code></pre></div>
<p>The available clusters are <code>C1</code>, <code>C2</code>, <code>C3</code> and <code>Tail</code>; <code>C1</code> is the clonal cluster and <code>C2</code> and <code>C3</code> are subclones.</p>
<p>To compute the dnds statistics one has to define a custom grouping on top of the actual clusters, so that the statistics are computed by group.</p>
<p>Groups are defined from a <code>mapping</code> variable.</p>
<blockquote>
<p>If <code>mapping  = c(</code>C1<code>= 'G1',</code>C2<code>= 'G1',</code>C3<code>= 'G1',</code>Tail<code>= 'G2')</code>, then mutations from clusters <code>C1</code>, <code>C2</code> and <code>C3</code> will be pooled into one group (<code>G1</code>), while mutations from cluster <code>Tail</code> will constitute a group themselves. By default, with <code>mapping = NULL</code>, each cluster is a group.</p>
</blockquote>
<p>One can now run the analysis using this grouping and the default gene list (<code>gene_list = NULL</code>) available in <code>dndscv</code>.</p>
<div class="sourceCode" id="cb4"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb4-1" data-line-number="1"><span class="co"># Run with this grouping and default gene list</span></a>
<a class="sourceLine" id="cb4-2" data-line-number="2">dnds_stats =<span class="st"> </span><span class="kw"><a href="../reference/dnds.html">dnds</a></span>(</a>
<a class="sourceLine" id="cb4-3" data-line-number="3">  clusters,</a>
<a class="sourceLine" id="cb4-4" data-line-number="4">  <span class="dt">mapping =</span> <span class="kw"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="st">`</span><span class="dt">C1</span><span class="st">`</span> =<span class="st"> 'Non-tail'</span>, <span class="st">`</span><span class="dt">C2</span><span class="st">`</span> =<span class="st"> 'Non-tail'</span>, <span class="st">`</span><span class="dt">C3</span><span class="st">`</span> =<span class="st"> 'Non-tail'</span>, <span class="st">`</span><span class="dt">Tail</span><span class="st">`</span> =<span class="st"> 'Tail'</span>),</a>
<a class="sourceLine" id="cb4-5" data-line-number="5">  <span class="dt">gene_list =</span> <span class="ot">NULL</span></a>
<a class="sourceLine" id="cb4-6" data-line-number="6">)</a>
<a class="sourceLine" id="cb4-7" data-line-number="7"><span class="co">#&gt; </span></a>
<a class="sourceLine" id="cb4-8" data-line-number="8"><span class="co">#&gt; =-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-</span></a>
<a class="sourceLine" id="cb4-9" data-line-number="9"><span class="co">#&gt;  MOBSTER wrapper for dndscv version 0.0.1.0</span></a>
<a class="sourceLine" id="cb4-10" data-line-number="10"><span class="co">#&gt; =-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-</span></a>
<a class="sourceLine" id="cb4-11" data-line-number="11"><span class="co">#&gt; Missing sample column.</span></a>
<a class="sourceLine" id="cb4-12" data-line-number="12"><span class="co">#&gt; &gt; Assuming these are mutations from a single patient, adding dummy sample id.</span></a>
<a class="sourceLine" id="cb4-13" data-line-number="13"><span class="co">#&gt; &gt; If this is not the case, label each mutation with a sample id.</span></a>
<a class="sourceLine" id="cb4-14" data-line-number="14"><span class="co">#&gt; </span></a>
<a class="sourceLine" id="cb4-15" data-line-number="15"><span class="co">#&gt;   Mutations  4643 </span></a>
<a class="sourceLine" id="cb4-16" data-line-number="16"><span class="co">#&gt;       Genes  0 </span></a>
<a class="sourceLine" id="cb4-17" data-line-number="17"><span class="co">#&gt;    Clusters  4 </span></a>
<a class="sourceLine" id="cb4-18" data-line-number="18"><span class="co">#&gt;     Samples  0 </span></a>
<a class="sourceLine" id="cb4-19" data-line-number="19"><span class="co">#&gt; Dnds groups  2 </span></a>
<a class="sourceLine" id="cb4-20" data-line-number="20"><span class="co">#&gt; </span></a>
<a class="sourceLine" id="cb4-21" data-line-number="21"><span class="co">#&gt; [refdb = hg19] </span></a>
<a class="sourceLine" id="cb4-22" data-line-number="22"><span class="co">#&gt; &gt; Removing chr from chromosome names for hg19 reference compatability</span></a>
<a class="sourceLine" id="cb4-23" data-line-number="23"><span class="co">#&gt; Mapping clusters to dnds_groups</span></a>
<a class="sourceLine" id="cb4-24" data-line-number="24"><span class="co">#&gt;   </span></a>
<a class="sourceLine" id="cb4-25" data-line-number="25"><span class="co">#&gt; Non-tail     Tail </span></a>
<a class="sourceLine" id="cb4-26" data-line-number="26"><span class="co">#&gt;     3637     1006 </span></a>
<a class="sourceLine" id="cb4-27" data-line-number="27"><span class="co">#&gt; </span></a>
<a class="sourceLine" id="cb4-28" data-line-number="28"><span class="co">#&gt; =-=-=-=-=-=-=-=-</span></a>
<a class="sourceLine" id="cb4-29" data-line-number="29"><span class="co">#&gt;  Running dndscv</span></a>
<a class="sourceLine" id="cb4-30" data-line-number="30"><span class="co">#&gt; =-=-=-=-=-=-=-=-</span></a>
<a class="sourceLine" id="cb4-31" data-line-number="31"><span class="co">#&gt; </span></a>
<a class="sourceLine" id="cb4-32" data-line-number="32"><span class="co">#&gt; dndscv @ dnds_group  Non-tail </span></a>
<a class="sourceLine" id="cb4-33" data-line-number="33"><span class="co">#&gt; </span></a>
<a class="sourceLine" id="cb4-34" data-line-number="34"><span class="co">#&gt; [1] Loading the environment...</span></a>
<a class="sourceLine" id="cb4-35" data-line-number="35"><span class="co">#&gt; [2] Annotating the mutations...</span></a>
<a class="sourceLine" id="cb4-36" data-line-number="36"><span class="co">#&gt; [3] Estimating global rates...</span></a>
<a class="sourceLine" id="cb4-37" data-line-number="37"><span class="co">#&gt; [4] Running dNdSloc...</span></a>
<a class="sourceLine" id="cb4-38" data-line-number="38"><span class="co">#&gt; [5] Running dNdScv...</span></a>
<a class="sourceLine" id="cb4-39" data-line-number="39"><span class="co">#&gt;     Regression model for substitutions (theta = 5.47).</span></a>
<a class="sourceLine" id="cb4-40" data-line-number="40"><span class="co">#&gt; </span></a>
<a class="sourceLine" id="cb4-41" data-line-number="41"><span class="co">#&gt; dndscv @ dnds_group  Tail </span></a>
<a class="sourceLine" id="cb4-42" data-line-number="42"><span class="co">#&gt; </span></a>
<a class="sourceLine" id="cb4-43" data-line-number="43"><span class="co">#&gt; [1] Loading the environment...</span></a>
<a class="sourceLine" id="cb4-44" data-line-number="44"><span class="co">#&gt; [2] Annotating the mutations...</span></a>
<a class="sourceLine" id="cb4-45" data-line-number="45"><span class="co">#&gt; [3] Estimating global rates...</span></a>
<a class="sourceLine" id="cb4-46" data-line-number="46"><span class="co">#&gt; [4] Running dNdSloc...</span></a>
<a class="sourceLine" id="cb4-47" data-line-number="47"><span class="co">#&gt; [5] Running dNdScv...</span></a>
<a class="sourceLine" id="cb4-48" data-line-number="48"><span class="co">#&gt;     Regression model for substitutions (theta = 0.000125).</span></a>
<a class="sourceLine" id="cb4-49" data-line-number="49"><span class="co">#&gt; Results: wall, wmis, wnon, wspl, wtru </span></a>
<a class="sourceLine" id="cb4-50" data-line-number="50"><span class="co">#&gt;  # A tibble: 10 x 5</span></a>
<a class="sourceLine" id="cb4-51" data-line-number="51"><span class="co">#&gt;    name           mle  cilow cihigh dnds_group</span></a>
<a class="sourceLine" id="cb4-52" data-line-number="52"><span class="co">#&gt;    &lt;chr&gt;        &lt;dbl&gt;  &lt;dbl&gt;  &lt;dbl&gt; &lt;chr&gt;     </span></a>
<a class="sourceLine" id="cb4-53" data-line-number="53"><span class="co">#&gt;  1 wmis  1.01         0.485    2.10 Non-tail  </span></a>
<a class="sourceLine" id="cb4-54" data-line-number="54"><span class="co">#&gt;  2 wnon  2.03         0.569    7.27 Non-tail  </span></a>
<a class="sourceLine" id="cb4-55" data-line-number="55"><span class="co">#&gt;  3 wspl  0.0000000298 0      Inf    Non-tail  </span></a>
<a class="sourceLine" id="cb4-56" data-line-number="56"><span class="co">#&gt;  4 wtru  1.39         0.417    4.61 Non-tail  </span></a>
<a class="sourceLine" id="cb4-57" data-line-number="57"><span class="co">#&gt;  5 wall  1.07         0.528    2.18 Non-tail  </span></a>
<a class="sourceLine" id="cb4-58" data-line-number="58"><span class="co">#&gt;  6 wmis  0.583        0.166    2.04 Tail      </span></a>
<a class="sourceLine" id="cb4-59" data-line-number="59"><span class="co">#&gt;  7 wnon  1.01         0.0860  11.8  Tail      </span></a>
<a class="sourceLine" id="cb4-60" data-line-number="60"><span class="co">#&gt;  8 wspl  0.0000000227 0      Inf    Tail      </span></a>
<a class="sourceLine" id="cb4-61" data-line-number="61"><span class="co">#&gt;  9 wtru  0.696        0.0694   6.99 Tail      </span></a>
<a class="sourceLine" id="cb4-62" data-line-number="62"><span class="co">#&gt; 10 wall  0.592        0.174    2.01 Tail      </span></a>
<a class="sourceLine" id="cb4-63" data-line-number="63"><span class="co">#&gt; Generating ouptut plot </span></a>
<a class="sourceLine" id="cb4-64" data-line-number="64"><span class="co">#&gt; </span></a></code></pre></div>
<p>The returned outputs are the tables returned by <code>dndscv</code>, as well as a <code>ggplot</code> plot for the object. The column <code>dnds_group</code> labels the group used to compute the statistics.</p>
<div class="sourceCode" id="cb5"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb5-1" data-line-number="1"><span class="co"># Summary statistics</span></a>
<a class="sourceLine" id="cb5-2" data-line-number="2"><span class="kw"><a href="https://rdrr.io/r/base/print.html">print</a></span>(dnds_stats<span class="op">$</span>dnds_summary)</a>
<a class="sourceLine" id="cb5-3" data-line-number="3"><span class="co">#&gt; # A tibble: 10 x 5</span></a>
<a class="sourceLine" id="cb5-4" data-line-number="4"><span class="co">#&gt;    name           mle  cilow cihigh dnds_group</span></a>
<a class="sourceLine" id="cb5-5" data-line-number="5"><span class="co">#&gt;    &lt;chr&gt;        &lt;dbl&gt;  &lt;dbl&gt;  &lt;dbl&gt; &lt;chr&gt;     </span></a>
<a class="sourceLine" id="cb5-6" data-line-number="6"><span class="co">#&gt;  1 wmis  1.01         0.485    2.10 Non-tail  </span></a>
<a class="sourceLine" id="cb5-7" data-line-number="7"><span class="co">#&gt;  2 wnon  2.03         0.569    7.27 Non-tail  </span></a>
<a class="sourceLine" id="cb5-8" data-line-number="8"><span class="co">#&gt;  3 wspl  0.0000000298 0      Inf    Non-tail  </span></a>
<a class="sourceLine" id="cb5-9" data-line-number="9"><span class="co">#&gt;  4 wtru  1.39         0.417    4.61 Non-tail  </span></a>
<a class="sourceLine" id="cb5-10" data-line-number="10"><span class="co">#&gt;  5 wall  1.07         0.528    2.18 Non-tail  </span></a>
<a class="sourceLine" id="cb5-11" data-line-number="11"><span class="co">#&gt;  6 wmis  0.583        0.166    2.04 Tail      </span></a>
<a class="sourceLine" id="cb5-12" data-line-number="12"><span class="co">#&gt;  7 wnon  1.01         0.0860  11.8  Tail      </span></a>
<a class="sourceLine" id="cb5-13" data-line-number="13"><span class="co">#&gt;  8 wspl  0.0000000227 0      Inf    Tail      </span></a>
<a class="sourceLine" id="cb5-14" data-line-number="14"><span class="co">#&gt;  9 wtru  0.696        0.0694   6.99 Tail      </span></a>
<a class="sourceLine" id="cb5-15" data-line-number="15"><span class="co">#&gt; 10 wall  0.592        0.174    2.01 Tail</span></a>
<a class="sourceLine" id="cb5-16" data-line-number="16"></a>
<a class="sourceLine" id="cb5-17" data-line-number="17"><span class="co"># Table observation countns</span></a>
<a class="sourceLine" id="cb5-18" data-line-number="18"><span class="kw"><a href="https://rdrr.io/r/base/print.html">print</a></span>(dnds_stats<span class="op">$</span>dndscv_table)</a>
<a class="sourceLine" id="cb5-19" data-line-number="19"><span class="co">#&gt; # A tibble: 40,182 x 15</span></a>
<a class="sourceLine" id="cb5-20" data-line-number="20"><span class="co">#&gt;    gene_name n_syn n_mis n_non n_spl wmis_cv wnon_cv wspl_cv pmis_cv ptrunc_cv</span></a>
<a class="sourceLine" id="cb5-21" data-line-number="21"><span class="co">#&gt;    &lt;chr&gt;     &lt;dbl&gt; &lt;dbl&gt; &lt;dbl&gt; &lt;dbl&gt;   &lt;dbl&gt;   &lt;dbl&gt;   &lt;dbl&gt;   &lt;dbl&gt;     &lt;dbl&gt;</span></a>
<a class="sourceLine" id="cb5-22" data-line-number="22"><span class="co">#&gt;  1 CNTN4         0     1     1     0    507.   3544.   3544. 1.36e-3 0.000170 </span></a>
<a class="sourceLine" id="cb5-23" data-line-number="23"><span class="co">#&gt;  2 P4HTM         0     0     1     0      0   11135.  11135. 9.67e-1 0.0000506</span></a>
<a class="sourceLine" id="cb5-24" data-line-number="24"><span class="co">#&gt;  3 PRSS46        0     1     0     0   4274.      0       0  1.39e-4 0.992    </span></a>
<a class="sourceLine" id="cb5-25" data-line-number="25"><span class="co">#&gt;  4 SPICE1        1     0     1     0      0    2859.   2859. 9.42e-1 0.000209 </span></a>
<a class="sourceLine" id="cb5-26" data-line-number="26"><span class="co">#&gt;  5 C3orf56       0     1     0     0   2721.      0       0  2.25e-4 0.993    </span></a>
<a class="sourceLine" id="cb5-27" data-line-number="27"><span class="co">#&gt;  6 CISH          0     1     0     0   2426.      0       0  2.54e-4 0.995    </span></a>
<a class="sourceLine" id="cb5-28" data-line-number="28"><span class="co">#&gt;  7 TMEM45A       0     1     0     0   1871.      0       0  3.36e-4 0.991    </span></a>
<a class="sourceLine" id="cb5-29" data-line-number="29"><span class="co">#&gt;  8 PLSCR1        0     1     0     0   1798.      0       0  3.50e-4 0.987    </span></a>
<a class="sourceLine" id="cb5-30" data-line-number="30"><span class="co">#&gt;  9 LMCD1         0     1     0     0   1757.      0       0  3.59e-4 0.991    </span></a>
<a class="sourceLine" id="cb5-31" data-line-number="31"><span class="co">#&gt; 10 DNAJC13       0     0     1     0      0    1711.   1711. 9.23e-1 0.000369 </span></a>
<a class="sourceLine" id="cb5-32" data-line-number="32"><span class="co">#&gt; # … with 40,172 more rows, and 5 more variables: pallsubs_cv &lt;dbl&gt;,</span></a>
<a class="sourceLine" id="cb5-33" data-line-number="33"><span class="co">#&gt; #   qmis_cv &lt;dbl&gt;, qtrunc_cv &lt;dbl&gt;, qallsubs_cv &lt;dbl&gt;, dnds_group &lt;chr&gt;</span></a>
<a class="sourceLine" id="cb5-34" data-line-number="34"></a>
<a class="sourceLine" id="cb5-35" data-line-number="35"><span class="co"># Plot</span></a>
<a class="sourceLine" id="cb5-36" data-line-number="36"><span class="kw"><a href="https://rdrr.io/r/base/print.html">print</a></span>(dnds_stats<span class="op">$</span>plot)</a></code></pre></div>
<p><img src="evolutionary_analysis_files/figure-html/unnamed-chunk-6-1.png" width="864"></p>
<p>By default, the plot contains all the results from the substitution models availavle in <code>dndscv</code>. If one wants to visualize only the global dnds score parameter <code>dndscv_plot</code> should be set to <code>wall</code> in the call to <code><a href="../reference/dnds.html">dnds()</a></code>.</p>
<div id="using-custom-genes-lists" class="section level3">
<h3 class="hasAnchor">
<a href="#using-custom-genes-lists" class="anchor"></a>Using custom genes lists</h3>
<p>A custom list of genes can be supplied in the call to <code><a href="../reference/dnds.html">dnds()</a></code> as the variable <code>genes_list</code>.</p>
<p>MOBSTER provides 4 lists of interests for this type of computation:</p>
<ul>
<li>a list of driver genes compiled in <code>Martincorena et al. Cell 171.5 (2017): 1029-1041.</code>;</li>
<li>a list of driver genes compiled in <code>Tarabichi, et al. Nature Genetics 50.12 (2018): 1630.</code>;</li>
<li>a list of essential genes compiled in <code>Wang et al. Science 350.6264 (2015): 1096-1101.</code>;</li>
<li>a list of essential genes compiled in <code>Bloomen et al. Science 350.6264 (2015): 1092-1096.</code>.</li>
</ul>
<p>which are available to load.</p>
<div class="sourceCode" id="cb6"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb6-1" data-line-number="1"><span class="co"># Load the list</span></a>
<a class="sourceLine" id="cb6-2" data-line-number="2"><span class="kw"><a href="https://rdrr.io/r/utils/data.html">data</a></span>(<span class="st">'cancer_genes_dnds'</span>, <span class="dt">package =</span> <span class="st">'mobster'</span>)</a>
<a class="sourceLine" id="cb6-3" data-line-number="3"></a>
<a class="sourceLine" id="cb6-4" data-line-number="4"><span class="co"># Each sublist is a list </span></a>
<a class="sourceLine" id="cb6-5" data-line-number="5"><span class="kw"><a href="https://rdrr.io/r/base/print.html">print</a></span>(<span class="kw"><a href="https://rdrr.io/r/base/lapply.html">lapply</a></span>(cancer_genes_dnds, head))</a>
<a class="sourceLine" id="cb6-6" data-line-number="6"><span class="co">#&gt; $Martincorena_drivers</span></a>
<a class="sourceLine" id="cb6-7" data-line-number="7"><span class="co">#&gt; [1] "CCDC6"     "EIF1AX"    "HIST1H2BD" "MED12"     "POLE"      "SMARCB1"  </span></a>
<a class="sourceLine" id="cb6-8" data-line-number="8"><span class="co">#&gt; </span></a>
<a class="sourceLine" id="cb6-9" data-line-number="9"><span class="co">#&gt; $Tarabichi_drivers</span></a>
<a class="sourceLine" id="cb6-10" data-line-number="10"><span class="co">#&gt; [1] "ACVR1"  "ACVR1B" "AKT1"   "ALK"    "AMER1"  "APC"   </span></a>
<a class="sourceLine" id="cb6-11" data-line-number="11"><span class="co">#&gt; </span></a>
<a class="sourceLine" id="cb6-12" data-line-number="12"><span class="co">#&gt; $Wang_essentials</span></a>
<a class="sourceLine" id="cb6-13" data-line-number="13"><span class="co">#&gt; [1] "ABL1"    "RPL23A"  "AARS2"   "TRMT112" "FARSA"   "ABCB7"  </span></a>
<a class="sourceLine" id="cb6-14" data-line-number="14"><span class="co">#&gt; </span></a>
<a class="sourceLine" id="cb6-15" data-line-number="15"><span class="co">#&gt; $Bloomen_essentials</span></a>
<a class="sourceLine" id="cb6-16" data-line-number="16"><span class="co">#&gt; [1] "AARS"     "AASDHPPT" "AATF"     "ABCB7"    "ABCE1"    "ABCF1"</span></a></code></pre></div>
<p>As suggestion, the list of driver genes by Tarabichi et al seems to work better with TCGA data. The two lists of essential genes have beenn instead compiled from two different cell lines.</p>
<p>To use a custom gene list run the following. Notice that if one uses data from a single patient it can often happen that there are not enough substitutions to compute the dnds values; in this case an error is returned which might be circumvented pooling data from multiple patients.</p>
<div class="sourceCode" id="cb7"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb7-1" data-line-number="1"><span class="co"># Not run here</span></a>
<a class="sourceLine" id="cb7-2" data-line-number="2">dnds_stats =<span class="st"> </span><span class="kw"><a href="../reference/dnds.html">dnds</a></span>(</a>
<a class="sourceLine" id="cb7-3" data-line-number="3">  clusters,</a>
<a class="sourceLine" id="cb7-4" data-line-number="4">  <span class="dt">mapping =</span> <span class="kw"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="st">`</span><span class="dt">C1</span><span class="st">`</span> =<span class="st"> 'Non-tail'</span>, <span class="st">`</span><span class="dt">C2</span><span class="st">`</span> =<span class="st"> 'Non-tail'</span>, <span class="st">`</span><span class="dt">C3</span><span class="st">`</span> =<span class="st"> 'Non-tail'</span>, <span class="st">`</span><span class="dt">Tail</span><span class="st">`</span> =<span class="st"> 'Tail'</span>),</a>
<a class="sourceLine" id="cb7-5" data-line-number="5">  <span class="dt">gene_list =</span> cancer_genes_dnds<span class="op">$</span>Martincorena_drivers</a>
<a class="sourceLine" id="cb7-6" data-line-number="6">)</a></code></pre></div>
</div>
<div id="pooling-data-from-multiple-patients" class="section level3">
<h3 class="hasAnchor">
<a href="#pooling-data-from-multiple-patients" class="anchor"></a>Pooling data from multiple patients</h3>
<p>The input format of the <code>dnds</code> function allows to pool data from several MOBSTER fits.</p>
<p>Here we load and pool together some of the real datasets available in the package, and use the same grouping as shown above.</p>
<div class="sourceCode" id="cb8"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb8-1" data-line-number="1"><span class="kw"><a href="https://rdrr.io/r/utils/data.html">data</a></span>(<span class="st">'LU4_lung_sample'</span>, <span class="dt">package =</span> <span class="st">'mobster'</span>)</a>
<a class="sourceLine" id="cb8-2" data-line-number="2"><span class="kw"><a href="https://rdrr.io/r/utils/data.html">data</a></span>(<span class="st">'LUFF76_lung_sample'</span>, <span class="dt">package =</span> <span class="st">'mobster'</span>)</a>
<a class="sourceLine" id="cb8-3" data-line-number="3"><span class="kw"><a href="https://rdrr.io/r/utils/data.html">data</a></span>(<span class="st">'PD4120a_breast_sample'</span>, <span class="dt">package =</span> <span class="st">'mobster'</span>)</a></code></pre></div>
<p>We pool the data before the call to <code>dnds</code>, subsetting the columns to the ones required for this computation.</p>
<div class="sourceCode" id="cb9"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb9-1" data-line-number="1">dnds_multi =<span class="st"> </span><span class="kw"><a href="../reference/dnds.html">dnds</a></span>(</a>
<a class="sourceLine" id="cb9-2" data-line-number="2">  <span class="kw"><a href="https://rdrr.io/r/base/cbind.html">rbind</a></span>(</a>
<a class="sourceLine" id="cb9-3" data-line-number="3">    <span class="kw"><a href="../reference/Clusters.html">Clusters</a></span>(LU4_lung_sample<span class="op">$</span>best) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">select</span>(chr, from, ref, alt, cluster) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">mutate</span>(<span class="dt">sample =</span> <span class="st">'LU4'</span>),</a>
<a class="sourceLine" id="cb9-4" data-line-number="4">    <span class="kw"><a href="../reference/Clusters.html">Clusters</a></span>(LUFF76_lung_sample<span class="op">$</span>best) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">select</span>(chr, from, ref, alt, cluster) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">mutate</span>(<span class="dt">sample =</span> <span class="st">'LUFF76'</span>),</a>
<a class="sourceLine" id="cb9-5" data-line-number="5">    <span class="kw"><a href="../reference/Clusters.html">Clusters</a></span>(PD4120a_breast_sample<span class="op">$</span>best) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">select</span>(chr, from, ref, alt, cluster) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">mutate</span>(<span class="dt">sample =</span> <span class="st">'PD4120a'</span>)</a>
<a class="sourceLine" id="cb9-6" data-line-number="6">  ),</a>
<a class="sourceLine" id="cb9-7" data-line-number="7">  <span class="dt">mapping =</span> <span class="kw"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="st">`</span><span class="dt">C1</span><span class="st">`</span> =<span class="st"> 'Non-tail'</span>, <span class="st">`</span><span class="dt">C2</span><span class="st">`</span> =<span class="st"> 'Non-tail'</span>, <span class="st">`</span><span class="dt">C3</span><span class="st">`</span> =<span class="st"> 'Non-tail'</span>, <span class="st">`</span><span class="dt">Tail</span><span class="st">`</span> =<span class="st"> 'Tail'</span>),</a>
<a class="sourceLine" id="cb9-8" data-line-number="8">)</a>
<a class="sourceLine" id="cb9-9" data-line-number="9"><span class="co">#&gt; </span></a>
<a class="sourceLine" id="cb9-10" data-line-number="10"><span class="co">#&gt; =-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-</span></a>
<a class="sourceLine" id="cb9-11" data-line-number="11"><span class="co">#&gt;  MOBSTER wrapper for dndscv version 0.0.1.0</span></a>
<a class="sourceLine" id="cb9-12" data-line-number="12"><span class="co">#&gt; =-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-</span></a>
<a class="sourceLine" id="cb9-13" data-line-number="13"><span class="co">#&gt; </span></a>
<a class="sourceLine" id="cb9-14" data-line-number="14"><span class="co">#&gt;   Mutations  8223 </span></a>
<a class="sourceLine" id="cb9-15" data-line-number="15"><span class="co">#&gt;       Genes  0 </span></a>
<a class="sourceLine" id="cb9-16" data-line-number="16"><span class="co">#&gt;    Clusters  4 </span></a>
<a class="sourceLine" id="cb9-17" data-line-number="17"><span class="co">#&gt;     Samples  3 </span></a>
<a class="sourceLine" id="cb9-18" data-line-number="18"><span class="co">#&gt; Dnds groups  2 </span></a>
<a class="sourceLine" id="cb9-19" data-line-number="19"><span class="co">#&gt; </span></a>
<a class="sourceLine" id="cb9-20" data-line-number="20"><span class="co">#&gt; [refdb = hg19] </span></a>
<a class="sourceLine" id="cb9-21" data-line-number="21"><span class="co">#&gt; &gt; Removing chr from chromosome names for hg19 reference compatability</span></a>
<a class="sourceLine" id="cb9-22" data-line-number="22"><span class="co">#&gt; Mapping clusters to dnds_groups</span></a>
<a class="sourceLine" id="cb9-23" data-line-number="23"><span class="co">#&gt;   </span></a>
<a class="sourceLine" id="cb9-24" data-line-number="24"><span class="co">#&gt; Non-tail     Tail </span></a>
<a class="sourceLine" id="cb9-25" data-line-number="25"><span class="co">#&gt;     5831     2392 </span></a>
<a class="sourceLine" id="cb9-26" data-line-number="26"><span class="co">#&gt; </span></a>
<a class="sourceLine" id="cb9-27" data-line-number="27"><span class="co">#&gt; =-=-=-=-=-=-=-=-</span></a>
<a class="sourceLine" id="cb9-28" data-line-number="28"><span class="co">#&gt;  Running dndscv</span></a>
<a class="sourceLine" id="cb9-29" data-line-number="29"><span class="co">#&gt; =-=-=-=-=-=-=-=-</span></a>
<a class="sourceLine" id="cb9-30" data-line-number="30"><span class="co">#&gt; </span></a>
<a class="sourceLine" id="cb9-31" data-line-number="31"><span class="co">#&gt; dndscv @ dnds_group  Non-tail </span></a>
<a class="sourceLine" id="cb9-32" data-line-number="32"><span class="co">#&gt; </span></a>
<a class="sourceLine" id="cb9-33" data-line-number="33"><span class="co">#&gt; [1] Loading the environment...</span></a>
<a class="sourceLine" id="cb9-34" data-line-number="34"><span class="co">#&gt; [2] Annotating the mutations...</span></a>
<a class="sourceLine" id="cb9-35" data-line-number="35"><span class="co">#&gt; [3] Estimating global rates...</span></a>
<a class="sourceLine" id="cb9-36" data-line-number="36"><span class="co">#&gt; [4] Running dNdSloc...</span></a>
<a class="sourceLine" id="cb9-37" data-line-number="37"><span class="co">#&gt; [5] Running dNdScv...</span></a>
<a class="sourceLine" id="cb9-38" data-line-number="38"><span class="co">#&gt;     Regression model for substitutions (theta = 10.6).</span></a>
<a class="sourceLine" id="cb9-39" data-line-number="39"><span class="co">#&gt; </span></a>
<a class="sourceLine" id="cb9-40" data-line-number="40"><span class="co">#&gt; dndscv @ dnds_group  Tail </span></a>
<a class="sourceLine" id="cb9-41" data-line-number="41"><span class="co">#&gt; </span></a>
<a class="sourceLine" id="cb9-42" data-line-number="42"><span class="co">#&gt; [1] Loading the environment...</span></a>
<a class="sourceLine" id="cb9-43" data-line-number="43"><span class="co">#&gt; [2] Annotating the mutations...</span></a>
<a class="sourceLine" id="cb9-44" data-line-number="44"><span class="co">#&gt; [3] Estimating global rates...</span></a>
<a class="sourceLine" id="cb9-45" data-line-number="45"><span class="co">#&gt; [4] Running dNdSloc...</span></a>
<a class="sourceLine" id="cb9-46" data-line-number="46"><span class="co">#&gt; [5] Running dNdScv...</span></a>
<a class="sourceLine" id="cb9-47" data-line-number="47"><span class="co">#&gt;     Regression model for substitutions (theta = 6.96).</span></a>
<a class="sourceLine" id="cb9-48" data-line-number="48"><span class="co">#&gt; Results: wall, wmis, wnon, wspl, wtru </span></a>
<a class="sourceLine" id="cb9-49" data-line-number="49"><span class="co">#&gt;  # A tibble: 10 x 5</span></a>
<a class="sourceLine" id="cb9-50" data-line-number="50"><span class="co">#&gt;    name           mle  cilow cihigh dnds_group</span></a>
<a class="sourceLine" id="cb9-51" data-line-number="51"><span class="co">#&gt;    &lt;chr&gt;        &lt;dbl&gt;  &lt;dbl&gt;  &lt;dbl&gt; &lt;chr&gt;     </span></a>
<a class="sourceLine" id="cb9-52" data-line-number="52"><span class="co">#&gt;  1 wmis  1.05         0.569    1.94 Non-tail  </span></a>
<a class="sourceLine" id="cb9-53" data-line-number="53"><span class="co">#&gt;  2 wnon  2.09         0.685    6.38 Non-tail  </span></a>
<a class="sourceLine" id="cb9-54" data-line-number="54"><span class="co">#&gt;  3 wspl  0.0000000160 0      Inf    Non-tail  </span></a>
<a class="sourceLine" id="cb9-55" data-line-number="55"><span class="co">#&gt;  4 wtru  1.13         0.396    3.23 Non-tail  </span></a>
<a class="sourceLine" id="cb9-56" data-line-number="56"><span class="co">#&gt;  5 wall  1.08         0.598    1.97 Non-tail  </span></a>
<a class="sourceLine" id="cb9-57" data-line-number="57"><span class="co">#&gt;  6 wmis  0.547        0.213    1.40 Tail      </span></a>
<a class="sourceLine" id="cb9-58" data-line-number="58"><span class="co">#&gt;  7 wnon  0.612        0.0649   5.77 Tail      </span></a>
<a class="sourceLine" id="cb9-59" data-line-number="59"><span class="co">#&gt;  8 wspl  0.0000000430 0      Inf    Tail      </span></a>
<a class="sourceLine" id="cb9-60" data-line-number="60"><span class="co">#&gt;  9 wtru  0.457        0.0522   4.00 Tail      </span></a>
<a class="sourceLine" id="cb9-61" data-line-number="61"><span class="co">#&gt; 10 wall  0.538        0.213    1.36 Tail      </span></a>
<a class="sourceLine" id="cb9-62" data-line-number="62"><span class="co">#&gt; Generating ouptut plot </span></a>
<a class="sourceLine" id="cb9-63" data-line-number="63"><span class="co">#&gt; </span></a>
<a class="sourceLine" id="cb9-64" data-line-number="64"></a>
<a class="sourceLine" id="cb9-65" data-line-number="65"><span class="kw"><a href="https://rdrr.io/r/base/print.html">print</a></span>(dnds_multi<span class="op">$</span>plot)</a></code></pre></div>
<p><img src="evolutionary_analysis_files/figure-html/unnamed-chunk-10-1.png" width="864"></p>
</div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="sidebar">

        <div id="tocnav">
      <h2 class="hasAnchor">
<a href="#tocnav" class="anchor"></a>Contents</h2>
      <ul class="nav nav-pills nav-stacked">
<li><a href="#computing-dnds-values">Computing dnds values</a></li>
      </ul>
</div>
      </div>

</div>



      <footer><div class="copyright">
  <p>Developed by Giulio Caravagna.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.4.1.</p>
</div>

      </footer>
</div>

  


  </body>
</html>
