<!-- Generated by pkgdown: do not edit by hand -->
<!DOCTYPE html>
<html lang="en">
  <head>
  <meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">

<title>Run a dN/dS analysis on MOBSTER clusters. — dnds • mobster</title>

<!-- favicons -->
<link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png">
<link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png">
<link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png" />
<link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png" />
<link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png" />
<link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png" />

<!-- jquery -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script>
<!-- Bootstrap -->

<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous" />

<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script>

<!-- bootstrap-toc -->
<link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script>

<!-- Font Awesome icons -->
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous" />
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous" />

<!-- clipboard.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script>

<!-- headroom.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script>

<!-- pkgdown -->
<link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script>




<meta property="og:title" content="Run a dN/dS analysis on MOBSTER clusters. — dnds" />
<meta property="og:description" content="This function takes a MOBSTER fit and runs `dndscv` (https://github.com/im3sanger/dndscv) to calculate dN/dS
values per cluster. It computes global dN/dS and per gene dN/dS values and makes a plot. dN/dS values are computed with
the best fitting MOBSTER model." />
<meta property="og:image" content="caravagn.github.io/mobster/logo.png" />




<!-- mathjax -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script>

<!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->



  </head>

  <body data-spy="scroll" data-target="#toc">
    <div class="container template-reference-topic">
      <header>
      <div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">mobster</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.1.1</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="../index.html">
    <span class="fas fa fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../articles/mobster.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="../articles/a1_introduction.html">1. Introduction</a>
    </li>
    <li>
      <a href="../articles/a2_plotting.html">2. Plotting fits</a>
    </li>
    <li>
      <a href="../articles/a3_bootstrap.html">3. Confidence estimation by the bootstrap</a>
    </li>
    <li>
      <a href="../articles/a4_popgen.html">4. Population Genetics statistics</a>
    </li>
    <li>
      <a href="../articles/a5_dnds.html">5. dN/dS statistics</a>
    </li>
    <li>
      <a href="../articles/a6_CloneTrees.html">6. Clone trees from clusters</a>
    </li>
    <li>
      <a href="../articles/a7_Example_tumour_simulation.html">7. Visualizing subclonal expansions</a>
    </li>
  </ul>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/caravagn/mobster/">
    <span class="fab fa fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
      
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header>

<div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Run a dN/dS analysis on MOBSTER clusters.</h1>
    <small class="dont-index">Source: <a href='https://github.com/caravagn/mobsterR/dnds.R'><code>R/dnds.R</code></a></small>
    <div class="hidden name"><code>dnds.Rd</code></div>
    </div>

    <div class="ref-description">
    <p>This function takes a MOBSTER fit and runs `dndscv` (https://github.com/im3sanger/dndscv) to calculate dN/dS
values per cluster. It computes global dN/dS and per gene dN/dS values and makes a plot. dN/dS values are computed with
the best fitting MOBSTER model.</p>
    </div>

    <pre class="usage"><span class='fu'>dnds</span>(
  <span class='kw'>x</span>,
  mapping = <span class='kw'>NULL</span>,
  gene_list = <span class='kw'>NULL</span>,
  colors = <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span>(Tail = <span class='st'>"gray"</span>),
  refdb = <span class='st'>"hg19"</span>,
  dndscv_plot = <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span>(<span class='st'>"wall"</span>, <span class='st'>"wmis"</span>, <span class='st'>"wnon"</span>, <span class='st'>"wspl"</span>, <span class='st'>"wtru"</span>),
  <span class='kw'>...</span>
)</pre>

    <h2 class="hasAnchor" id="arguments"><a class="anchor" href="#arguments"></a>Arguments</h2>
    <table class="ref-arguments">
    <colgroup><col class="name" /><col class="desc" /></colgroup>
    <tr>
      <th>x</th>
      <td><p>A MOBSTER fit object.</p></td>
    </tr>
    <tr>
      <th>mapping</th>
      <td><p>The groups used to compute this statistics are defined by this variable. If
`mapping  = c(`A` = 'G1', `B` = 'G1', `C` = 'G2')`, then mutations from clusters `A` and 
`B` will be pooled into one group (`G1`), while mutations from cluster `C` will constitute
a group themselves. By default, with `mapping = NULL`, each cluster is a group.</p></td>
    </tr>
    <tr>
      <th>gene_list</th>
      <td><p>An optional vector of gene names to infer dN/dS values,
default (`NULL`) is to use <code>dndscv</code> default (whole-exome. This package provides lists
genes that can be used for this value (essential genes, cancer genes, etc.); see package data.</p></td>
    </tr>
    <tr>
      <th>colors</th>
      <td><p>If provided, these colours will be used for each cluster.
If a subset of colours is provided, palette Set1 from <code>RColorBrewer</code> is used.
By default the tail colour is provided as 'gainsboro'.</p></td>
    </tr>
    <tr>
      <th>refdb</th>
      <td><p>The genome referene to use, default is to use hg19. Other references are available from
https://github.com/im3sanger/dndscv_data</p></td>
    </tr>
    <tr>
      <th>dndscv_plot</th>
      <td><p>What of the dndscv scores should be visualized in a plot, by default all the statistcs
are reported. One can use `dndscv_plot = wall` to get only the global dnds value.</p></td>
    </tr>
    <tr>
      <th>...</th>
      <td><p>Extra parameters forwarded to <code>dndscv</code>.</p></td>
    </tr>
    </table>

    <h2 class="hasAnchor" id="value"><a class="anchor" href="#value"></a>Value</h2>

    <p>The fit object is a list with the summary table and the observation counts reported
by package <code>dndscv</code>, together with a <code>ggplot</code> plot for the results.</p>

    <h2 class="hasAnchor" id="examples"><a class="anchor" href="#examples"></a>Examples</h2>
    <pre class="examples"><div class='input'>
<span class='co'># Example run with real data</span>
<span class='fu'><a href='https://rdrr.io/r/utils/data.html'>data</a></span>(<span class='st'>'LUFF76_lung_sample'</span>, package = <span class='st'>'mobster'</span>)

<span class='kw'>clusters</span> <span class='op'>=</span> <span class='fu'><a href='Clusters.html'>Clusters</a></span>(<span class='kw'>LUFF76_lung_sample</span><span class='op'>$</span><span class='kw'>best</span>)

<span class='kw'>dnds_stats</span> <span class='op'>=</span> <span class='fu'>dnds</span>(<span class='kw'>clusters</span>, gene_list = <span class='kw'>NULL</span>)
</div><div class='output co'>#&gt; <span class='message'>Missing 'sample' column, assuming mutations from a single patient (adding a sample label otherwise).</span></div><div class='output co'>#&gt; <span class='warning'>Warning: Unknown or uninitialised column: `sample`.</span></div><div class='output co'>#&gt; <span class='message'><span style='color: #00BBBB;'>ℹ</span><span> 2298 mutations; </span><span style='color: #00BB00;'>'by cluster'</span><span> groups in 0 samples, with </span><span style='color: #00BB00;'>no genes (default dndscv)</span><span>.</span></div><div class='output co'>#&gt; <span class='message'>[refdb = hg19] Removing chr from chromosome names for hg19 reference compatability</span></div><div class='output co'>#&gt; 
#&gt;   C1 Tail 
#&gt; 1222 1076 </div><div class='output co'>#&gt; </div><div class='output co'>#&gt; <span class='message'></span><span style='color: #00BBBB;'>──</span><span> </span><span style='font-weight: bold;'>Running dndscv</span><span> </span><span style='color: #00BBBB;'>────────────────────────────────────────────────────────────────────────────────────</span><span></span></div><div class='output co'>#&gt; </div><div class='output co'>#&gt; <span class='message'>── Group Tail </span></div><div class='output co'>#&gt; <span class='warning'>Warning: replacing previous import ‘Biostrings::translate’ by ‘seqinr::translate’ when loading ‘dndscv’</span></div><div class='output co'>#&gt; <span class='message'>[1] Loading the environment...</span></div><div class='output co'>#&gt; <span class='message'>[2] Annotating the mutations...</span></div><div class='output co'>#&gt; <span class='warning'>Warning: Mutations observed in contiguous sites within a sample. Please annotate or remove dinucleotide or complex substitutions for best results.</span></div><div class='output co'>#&gt; <span class='message'>[3] Estimating global rates...</span></div><div class='output co'>#&gt; <span class='warning'>Warning: glm.fit: fitted rates numerically 0 occurred</span></div><div class='output co'>#&gt; <span class='warning'>Warning: glm.fit: fitted rates numerically 0 occurred</span></div><div class='output co'>#&gt; <span class='warning'>Warning: glm.fit: fitted rates numerically 0 occurred</span></div><div class='output co'>#&gt; <span class='message'>[4] Running dNdSloc...</span></div><div class='output co'>#&gt; <span class='message'>[5] Running dNdScv...</span></div><div class='output co'>#&gt; <span class='message'>    Regression model for substitutions (theta = 6.69e-05).</span></div><div class='output co'>#&gt; </div><div class='output co'>#&gt; <span class='message'>── Group C1 </span></div><div class='output co'>#&gt; <span class='message'>[1] Loading the environment...</span></div><div class='output co'>#&gt; <span class='message'>[2] Annotating the mutations...</span></div><div class='output co'>#&gt; <span class='warning'>Warning: Mutations observed in contiguous sites within a sample. Please annotate or remove dinucleotide or complex substitutions for best results.</span></div><div class='output co'>#&gt; <span class='message'>[3] Estimating global rates...</span></div><div class='output co'>#&gt; <span class='warning'>Warning: glm.fit: algorithm did not converge</span></div><div class='output co'>#&gt; <span class='warning'>Warning: glm.fit: fitted rates numerically 0 occurred</span></div><div class='output co'>#&gt; <span class='warning'>Warning: glm.fit: algorithm did not converge</span></div><div class='output co'>#&gt; <span class='warning'>Warning: glm.fit: fitted rates numerically 0 occurred</span></div><div class='output co'>#&gt; <span class='warning'>Warning: glm.fit: algorithm did not converge</span></div><div class='output co'>#&gt; <span class='warning'>Warning: glm.fit: fitted rates numerically 0 occurred</span></div><div class='output co'>#&gt; <span class='message'>[4] Running dNdSloc...</span></div><div class='output co'>#&gt; <span class='message'>[5] Running dNdScv...</span></div><div class='output co'>#&gt; </span><span style='color: #BB0000;'>dndscv error
#&gt; </span><span style='color: #0000BB;'>Error in while ((it &lt;- it + 1) &lt; limit &amp;&amp; abs(del) &gt; eps) {: missing value where TRUE/FALSE needed
#&gt; </span><span></div><div class='output co'>#&gt; <span class='message'>── dndscv results ──────────────────────────────────────────────────── wall, wmis, wnon, wspl, wtru ──</span></div><div class='output co'>#&gt; </span><span style='color: #555555;'># A tibble: 5 x 5</span><span>
#&gt;   name            mle cilow cihigh dnds_group
#&gt;   </span><span style='color: #555555;font-style: italic;'>&lt;chr&gt;</span><span>         </span><span style='color: #555555;font-style: italic;'>&lt;dbl&gt;</span><span> </span><span style='color: #555555;font-style: italic;'>&lt;dbl&gt;</span><span>  </span><span style='color: #555555;font-style: italic;'>&lt;dbl&gt;</span><span> </span><span style='color: #555555;font-style: italic;'>&lt;chr&gt;</span><span>     
#&gt; </span><span style='color: #555555;'>1</span><span> wmis  0.731         0.113   4.73 Tail      
#&gt; </span><span style='color: #555555;'>2</span><span> wnon  0.000</span><span style='text-decoration: underline;'>000</span><span>008</span><span style='text-decoration: underline;'>04</span><span> 0     </span><span style='color: #BB0000;'>Inf</span><span>    Tail      
#&gt; </span><span style='color: #555555;'>3</span><span> wspl  0.000</span><span style='text-decoration: underline;'>000</span><span>026</span><span style='text-decoration: underline;'>0</span><span>  0     </span><span style='color: #BB0000;'>Inf</span><span>    Tail      
#&gt; </span><span style='color: #555555;'>4</span><span> wtru  0.000</span><span style='text-decoration: underline;'>000</span><span>017</span><span style='text-decoration: underline;'>0</span><span>  0     </span><span style='color: #BB0000;'>Inf</span><span>    Tail      
#&gt; </span><span style='color: #555555;'>5</span><span> wall  0.701         0.109   4.51 Tail      </div></span></pre>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top">
      <h2 data-toc-skip>Contents</h2>
    </nav>
  </div>
</div>


      <footer>
      <div class="copyright">
  <p>Developed by <a href='https://sites.google.com/site/giuliocaravagna/'>Giulio Caravagna</a>.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.6.0.</p>
</div>

      </footer>
   </div>

  


  </body>
</html>


